{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combining all downstream used A2TEA data into one RDS object\n",
    "## Also performing necessary data manipulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "###packages currently installed via conda\n",
    "\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(version = \"3.12\")\n",
    "\n",
    "#BiocManager::install(\"DESeq2\")\n",
    "#BiocManager::install(\"ggtree\")\n",
    "#BiocManager::install(\"Biostrings\")\n",
    "\n",
    "#lfcShrinkage isn't installable at the moment - problem for later\n",
    "#install.packages(\"mvtnorm\")\n",
    "#BiocManager::install(\"apeglm\")\n",
    "\n",
    "#install.packages(\"data.table\")\n",
    "\n",
    "#install.packages(\"seqinr\")\n",
    "#install.packages(\"VennDiagram\")\n",
    "#install.packages(\"UpSetR\")\n",
    "#install.packages(\"cowplot\")\n",
    "#install.packages(\"ggplotify\")\n",
    "\n",
    "###############################################\n",
    "\n",
    "\n",
    "if (!requireNamespace(\"BiocManager\", quietly=TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install(\"ggtree\")\n",
    "BiocManager::install(\"ggtreeExtra\")\n",
    "BiocManager::install(\"Biostrings\")\n",
    "\n",
    "install.packages(\"VennDiagram\")\n",
    "install.packages(\"UpSetR\")\n",
    "install.packages(\"cowplot\")\n",
    "install.packages(\"ggplotify\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(DESeq2)\n",
    "#library(BiocGenerics) # comes with DESeq2\n",
    "#library(apeglm) # currently difficulties with install # better DESseq2 calc; later \n",
    "\n",
    "library(tidyverse)\n",
    "library(ggtree)\n",
    "library(Biostrings)\n",
    "library(seqinr)\n",
    "library(VennDiagram)\n",
    "library(UpSetR)\n",
    "library(cowplot)\n",
    "library(ggplotify)\n",
    "library(ape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Find all DESeq2 differential expression RDS objects and load them\n",
    "##### Names are given based on the species/ecotype/etc. name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dea_list <- list.files(path = \"R/deseq2/dea_final/\", pattern = \"dea_*\", full.names=TRUE)\n",
    "dea_list_short <- list.files(path = \"R/deseq2/dea_final/\", pattern = \"dea_*\", full.names=FALSE)\n",
    "\n",
    "listRDS <- lapply(dea_list, readRDS, .GlobalEnv)\n",
    "\n",
    "for (i in 1:length(listRDS)){\n",
    "    assign(str_sub(dea_list_short[[i]], start=5), listRDS[[i]])\n",
    "}\n",
    "\n",
    "# remove unused combined list object\n",
    "rm(listRDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#nice for later - but needs install of package \"apeglm\"\n",
    "\n",
    "#link on logFC shrinking\n",
    "#https://support.bioconductor.org/p/77461/\n",
    "#?lfcShrink\n",
    "#lfcShrink(arabidopsis_dea, coef=2, type=\"apeglm\")\n",
    "#resultsNames(arabidopsis_dea)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DESeq2 testing\n",
    "#At <- readRDS(\"R/deseq2/dea_final/dea_Arabidopsis_thaliana\")\n",
    "#DESeq2::results(At)\n",
    "#as.data.frame(results(Arabidopsis_thaliana))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_list <- vector()\n",
    "\n",
    "for (i in 1:length(dea_list_short)){\n",
    "    species_list <- c(species_list, str_sub(dea_list_short[[i]], start=5))\n",
    "}\n",
    "\n",
    "list_AllSpeciesDEResultsDataFrames <- list()\n",
    "\n",
    "for (i in species_list) {\n",
    "    # individual species file could be deleted but perhaps I'll need them later; kept for now...\n",
    "    assign(paste0(i, \"_DEresultsTable\"), as.data.frame(results(get(i))))\n",
    "    # create gene name vector here (when list is constructed rownames become unique species-gene combination)\n",
    "    gene <- rownames(get(paste0(i, \"_DEresultsTable\")))\n",
    "    df_with_genes <- add_column(get(paste0(i, \"_DEresultsTable\")), gene, .before = \"baseMean\")\n",
    "    # also add a species column which will will also come in handy during the shiny steps\n",
    "    species <- replicate(nrow(df_with_genes), i)\n",
    "    df_with_genes <- add_column(df_with_genes, species, .before = \"gene\")\n",
    "    # create list of dataframes, which will come in handy\n",
    "    list_AllSpeciesDEResultsDataFrames[[i]] <- df_with_genes\n",
    "    # list_AllSpeciesDEResultsDataFrames[[i]] <- (get(paste0(i, \"_DEresultsTable\")))\n",
    "}\n",
    "\n",
    "#str(list_AllSpeciesDEResultsDataFrames)\n",
    "\n",
    "#create combined \"mega\" dataframe of all species, which is going  to be used for shiny lookup\n",
    "\n",
    "combined_AllSpeciesDEResultsDataFrames <- do.call(\"rbind\", list_AllSpeciesDEResultsDataFrames)\n",
    "\n",
    "\n",
    "#str(combined_AllSpeciesDEResultsDataFrames)\n",
    "#head(combined_AllSpeciesDEResultsDataFrames)\n",
    "#head(list_AllSpeciesDEResultsDataFrames[[1]])\n",
    "#?rbind"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### creating long format HOG-genes relation table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HOG_file_raw <- read_delim(\"orthofinder/final-results/Phylogenetic_Hierarchical_Orthogroups/N0.tsv\", delim = \"\\t\")\n",
    "\n",
    "#drop the OG, Gene Tree Parent Clade\n",
    "HOG_file_short <- HOG_file_raw %>% select (-c(OG, `Gene Tree Parent Clade`)) \n",
    "\n",
    "\n",
    "#merge all species columns into one and remove the solo ones\n",
    "#use all colnames except first - thus all species\n",
    "HOG_file_merged <- HOG_file_short %>% unite(x,\n",
    "                                          c(colnames(HOG_file_short)[-1]),\n",
    "                                          sep = \",\", \n",
    "                                          na.rm = TRUE,\n",
    "                                          remove = TRUE)\n",
    "\n",
    "#transform into long format - each row 1 gene and it's corresponding HOG in seperate columns\n",
    "HOG_file_long <- HOG_file_merged %>%\n",
    "                    mutate(unpacked = str_split(x, \",\")) %>%\n",
    "                    unnest(cols = c(unpacked)) %>%\n",
    "                    mutate(genes = str_trim(unpacked)) %>% \n",
    "                    select(-c(x, unpacked)) %>% \n",
    "                    rename(gene = genes)\n",
    "\n",
    "str(HOG_file_long)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### adding specific HOG or singleton info as HOG column to DE tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#?full_join\n",
    "HOG_DE.a2tea <- full_join(combined_AllSpeciesDEResultsDataFrames, HOG_file_long, \n",
    "              by = c(\"gene\")) %>% \n",
    "                replace_na(list(HOG = \"singleton\"))\n",
    "\n",
    "#add column for significance - default level is padj <= 0.1\n",
    "\n",
    "#subset tidyverse with the right functions ;D\n",
    "#filter(HOG_DE.a2tea, padj < 0.05)\n",
    "#?drop_na\n",
    "\n",
    "significant <- vector()\n",
    "for (FDR in HOG_DE.a2tea$padj) {\n",
    "    if (!is.na(FDR) && FDR < 0.1) {\n",
    "        significant <- c(significant, \"yes\")\n",
    "    }\n",
    "    else if (!is.na(FDR) && FDR > 0.1) {\n",
    "        significant <- c(significant, \"no\")\n",
    "    }\n",
    "    else if (is.na(FDR)) {\n",
    "        significant <- c(significant, \"no\")\n",
    "    }\n",
    "}\n",
    "\n",
    "HOG_DE.a2tea <- add_column(HOG_DE.a2tea, significant, .after = \"padj\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#?full_join"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load in the rest of the data - hypotheses, trees, fasta and msa for the start \n",
    "### + toDo general stats, especially once Orthofinder calculates for HOG)\n",
    "### +toDo Venn DIagrammes (until 5 species or so then alternative plots)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#extended_BLAST_hits <- readRDS(\"tea/2/extended_BLAST_hits/extended_BLAST_hits.RDS\")\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    nrow_table=\"numeric\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\")\n",
    "         )\n",
    "#expanded_OGs_short\n",
    "extended_BLAST_hits[[\"N0.HOG0000207\"]]@nrow_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hypotheses <- read_delim(\"hypotheses.tsv\", delim = \"\\t\")\n",
    "\n",
    "\n",
    "#create hypotheses object\n",
    "# each object has list of exp. OGs\n",
    "#iterate over list of hyptheses and associate expansions, fasta, msa and trees\n",
    "\n",
    "\n",
    "#first goal - hypothesis object with hypothesis and all exp.HOGs\n",
    "\n",
    "#could do it over objects in tea directory\n",
    "#list.files(path = \"tea\", pattern = \"*\", full.names=FALSE)\n",
    "\n",
    "\n",
    "#alternatively via parsing of hyptheses.tsv\n",
    "hypotheses$hypothesis\n",
    "\n",
    "\n",
    "## define three classes\n",
    "# class for the expanded_OG - containing all different types of data we have on it\n",
    "setClass(\"expanded_OG\", slots=list(genes=\"spec_tbl_df\",\n",
    "                                   blast_table=\"tbl_df\",\n",
    "                                   nrow_table=\"numeric\",\n",
    "                                   num_genes_HOG=\"numeric\",\n",
    "                                   num_genes_extend=\"numeric\",\n",
    "                                   num_genes_complete=\"numeric\",\n",
    "                                   genes_HOG=\"tbl_df\",\n",
    "                                   fasta_files=\"list\", \n",
    "                                   msa=\"AAMultipleAlignment\", \n",
    "                                   tree=\"phylo\"))\n",
    "\n",
    "\n",
    "# class for the hypotheses\n",
    "# adding a prototype is essential here for ortho_intersect_plot since gg has to be defined as \n",
    "# register an old-style S3 class using setOldClass\n",
    "#https://stackoverflow.com/questions/12636056/why-sometimes-i-cant-set-a-class-definition-as-slot-in-a-s4-class\n",
    "#https://stackoverflow.com/questions/13841400/use-s3-virtual-class-as-slot-of-an-s4-class-got-error-got-class-s4-should-b\n",
    "#had trouble getting this to work; thus solved differently\n",
    "# -> seperate list object \"Ortho_intersect_plots\" hypothesis specific access via index\n",
    "setClass(\"hypothesis\", \n",
    "#         prototype=prototype(ortho_intersect_plot=structure(list(), class=\"gList\")),\n",
    "         slots=list(description=\"character\", \n",
    "                                  number=\"character\",\n",
    "                                  expanded_in =\"character\", \n",
    "                                  compared_to=\"character\", \n",
    "                                  expanded_OGs=\"list\",\n",
    "                                  species_tree=\"phylo\"))\n",
    "#                                  ortho_intersect_plot=\"gg\"))\n",
    "\n",
    "# class for extended BLAST hits info\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    nrow_table=\"numeric\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\")\n",
    "         )\n",
    "\n",
    "\n",
    "rm(t)\n",
    "rm(hypothesis_1)\n",
    "rm(hypothesis_2)\n",
    "rm(HYPOTHESES.a2tea)\n",
    "\n",
    "\n",
    "#remove protein_names in the snakemake pipeline - directories clean enough\n",
    "\n",
    "\n",
    "for (hypothesis in hypotheses$hypothesis) {\n",
    "    # read-in extended_BLAST_hits.RDS object of hypothesis\n",
    "    extended_BLAST_hits <- readRDS(paste0(\"tea/\", hypothesis, \"/extended_BLAST_hits/extended_BLAST_hits.RDS\"))\n",
    "    \n",
    "    # print(paste0(\"hypothesis_\", hypo))\n",
    "    \n",
    "    # create empty list object for hypothesis #\n",
    "    assign(paste0(\"hypothesis_\", hypothesis), list())\n",
    "\n",
    "    # assign list of names\n",
    "    expanded_OGs <- list.files(path = paste0(\"tea/\", hypothesis, \"/exp_OGs_proteinnames/\"), \n",
    "                               pattern = \"*\", \n",
    "                               full.names=FALSE)\n",
    "    expanded_OGs_short <- str_sub(expanded_OGs, end=-5)\n",
    "\n",
    "    # assign or subset the orthofinder species_tree based on the current hypothesis\n",
    "    speciesTree <- ape::keep.tip(read.tree(\"orthofinder/final-results/Species_Tree/SpeciesTree_rooted_node_labels.txt\"), \n",
    "                                 c(unlist(str_split(hypotheses$expanded_in[hypothesis], \";\")), \n",
    "                                   unlist(str_split(hypotheses$compared_to[hypothesis], \";\"))))\n",
    "    \n",
    "    \n",
    "    #create empty list\n",
    "    t <- list()\n",
    "\n",
    "    for (exp_OG in expanded_OGs_short) {\n",
    "    # print(exp_OG)\n",
    "        test <- new(\"expanded_OG\", \n",
    "             genes=read_table(paste0(\"tea/\", hypothesis, \"/exp_OGs_proteinnames/\", exp_OG, \".txt\"), col_names = FALSE),\n",
    "             blast_table=extended_BLAST_hits[[exp_OG]]@blast_table,\n",
    "             num_genes_HOG=extended_BLAST_hits[[exp_OG]]@num_genes_HOG,\n",
    "             num_genes_extend=extended_BLAST_hits[[exp_OG]]@num_genes_extend,\n",
    "             num_genes_complete=extended_BLAST_hits[[exp_OG]]@num_genes_complete,\n",
    "             genes_HOG=extended_BLAST_hits[[exp_OG]]@genes_HOG,\n",
    "             fasta_files=read.fasta(paste0(\"tea/\", hypothesis, \"/fa_records/\", exp_OG,\".fa\"), seqtype = \"AA\", as.string = TRUE), \n",
    "             msa=readAAMultipleAlignment(paste0(\"tea/\", hypothesis, \"/muscle/\", exp_OG, \".afa\")), \n",
    "             tree=read.tree(paste0(\"tea/\", hypothesis, \"/trees/\", exp_OG, \".tree\")))\n",
    "   \n",
    "        x <- list(test)\n",
    "        names(x) <- paste0(exp_OG)\n",
    "        t <- c(t, x)\n",
    "    }\n",
    " \n",
    "    # create for each hypothesis a complete hypothesis object with correct naming\n",
    "    # order: hypothesis@info (name, number, expand, compared, exp OGs)$indiv. exp_OGs@objects (exp.OG list, fa, msa, tree)\n",
    "    # \"expanded_in\" and \"compared_to\" are split on \";\" in case their size is > 1\n",
    "    assign(paste0(\"hypothesis_\", hypothesis),\n",
    "              new(\"hypothesis\", \n",
    "                  description=hypotheses$name[hypothesis],\n",
    "                  number=as.character(hypothesis),\n",
    "                  expanded_in=unlist(str_split(hypotheses$expanded_in[hypothesis], \";\")),\n",
    "                  compared_to=unlist(str_split(hypotheses$compared_to[hypothesis], \";\")), \n",
    "                  expanded_OGs=t,\n",
    "                  species_tree=speciesTree))#,\n",
    "#                  ortho_intersect_plot=NULL))\n",
    "}\n",
    "\n",
    "\n",
    "# create empty list for final complete hypothesis object\n",
    "\n",
    "#rm\n",
    "\n",
    "# create name list of all hypotheses\n",
    "hypotheses_list <- ls(pattern = \"hypothesis_\")\n",
    "\n",
    "# create empty list object to completely hold all hypotheses and associated data\n",
    "HYPOTHESES.a2tea <- list()\n",
    "\n",
    "for (hypothesis in hypotheses_list) {\n",
    "    h <- list(get(hypothesis))\n",
    "    names(h) <- paste0(hypothesis)\n",
    "    HYPOTHESES.a2tea <- c(HYPOTHESES.a2tea, h)\n",
    "}\n",
    "\n",
    "rm(hypothesis_1)\n",
    "rm(hypothesis_2)\n",
    "\n",
    "\n",
    "# final object is called:\n",
    "# HYPOTHESES.a2tea\n",
    "\n",
    "\n",
    "#add expanded OGs as column\n",
    "#    add_column(get(paste0(\"hypothesis_\", hypo)), expanded_OGs)\n",
    "#class(hypothesis_1@expanded_OGs) <- c(\"expanded_OG\", class(hypothesis_1@expanded_OGs))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "length(hypotheses$hypothesis)\n",
    "length(HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000027@genes$X1)\n",
    "length(HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000027@fasta_files)\n",
    "HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000028@blast_table\n",
    "#?setClass"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### HOG level tables -needs to be behind creation of hypotheses, because they are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create hypothesis specfic HOG level file\n",
    "\n",
    "#find all tibbles -list\n",
    "#list.files(path = \"tea/1/expansion_tibble\", pattern = \"expansion_tibble.rds\", full.names=TRUE)\n",
    "\n",
    "\n",
    "#length(hypotheses_list)\n",
    "\n",
    "\n",
    "HOG_level_list <- list()\n",
    "\n",
    "#for (i in 1:length(hypotheses_list)) {\n",
    "for (i in 1:length(hypotheses$hypothesis)) {\n",
    "    assign(paste0(\"hypothesis_\", i, \"_expansion_tibble\"),\n",
    "           readRDS(\n",
    "               list.files(path = paste0(\"tea/\", i, \"/expansion_tibble\"), \n",
    "                      pattern = \"expansion_tibble.rds\", \n",
    "                      full.names=TRUE)\n",
    "                  )\n",
    "           )\n",
    "    # add to list and name list entries according to hypothesis\n",
    "    hog_level <- list(get(paste0(\"hypothesis_\", i, \"_expansion_tibble\")))\n",
    "    names(hog_level) <- paste0(\"hypothesis_\", i)\n",
    "    HOG_level_list <- c(HOG_level_list, hog_level)\n",
    "}\n",
    "\n",
    "#e.g. the following displays the tibble\n",
    "#hypothesis_2_expansion_tibble\n",
    "\n",
    "#r\n",
    "#append \"_gene_count\" to all species in HOG tables\n",
    "#get number of significantly regulated genes per HOG (per species and total) \n",
    "# -> (hmm perhaps do this once and then merge?)\n",
    "\n",
    "\n",
    "# get count of species, HOG, significant combination from HOG_DE.a2tea\n",
    "# reduce it to HOG, species, count\n",
    "sig_genes_per_species_and_HOG <- HOG_DE.a2tea %>% \n",
    "                                     filter(significant == c(\"yes\")) %>%\n",
    "                                     filter(HOG != c(\"singleton\")) %>%\n",
    "                                        group_by(HOG, species, significant) %>% \n",
    "                                        mutate(count = n()) %>%\n",
    "                                        ungroup() %>%\n",
    "                                        select(HOG, species, count) %>%\n",
    "            # https://www.r-bloggers.com/2018/05/workaround-for-tidyrspread-with-duplicate-row-identifiers/\n",
    "            # spread error when no indexing for data\n",
    "                                            group_by(species) %>% \n",
    "                                            mutate(grouped_id = row_number()) %>%  \n",
    "                                            spread(species, count) %>% \n",
    "                                            select(-grouped_id) %>%\n",
    "            # easy workaround for duplicated rows\n",
    "                                                distinct()%>% \n",
    "            # rename species columns containing now the counts of sig. DE genes\n",
    "                                                    rename_at(vars(-HOG), ~ paste0(., '_sigDE'))\n",
    "    # and also add a column summing the rowwise sig. DE counts for all species\n",
    "    total_sigDE <- sig_genes_per_species_and_HOG %>% select(-HOG) %>% rowSums(na.rm = TRUE) \n",
    "    sig_genes_per_species_and_HOG <- add_column(sig_genes_per_species_and_HOG, total_sigDE)\n",
    "    sig_genes_per_species_and_HOG <- sig_genes_per_species_and_HOG  %>%\n",
    "                                         group_by(HOG) %>%\n",
    "                                         # mutate all NAs to 0s\n",
    "                                         mutate_at(vars(-group_cols()), ~replace(., is.na(.), 0)) %>%\n",
    "                                         # merge rows per HOG - results in one line per HOG\n",
    "                                         #summarise_all(funs(sum))\n",
    "                                         summarise_all(list(sum))\n",
    "\n",
    "# change the zeros back to NAs\n",
    "    sig_genes_per_species_and_HOG <- sig_genes_per_species_and_HOG  %>%\n",
    "                                         na_if(0)\n",
    "\n",
    "#sig_genes_per_species_and_HOG\n",
    "#total_sigDE\n",
    "#inner_join(HOG_level_list[[1]], sig_genes_per_species_and_HOG, by = c(\"HOG\"))\n",
    "#full_join important - better to keep all HOG rows and allow filtering dynamically in the WebAPP itself\n",
    "\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "    HOG_level_list[[i]] <- HOG_level_list[[i]] %>% rename_at(vars(-HOG, -expansion), ~ paste0(., '_total'))\n",
    "    HOG_level_list[[i]] <- full_join(HOG_level_list[[i]], sig_genes_per_species_and_HOG, by = c(\"HOG\"))\n",
    "}\n",
    "\n",
    "str(HOG_level_list)\n",
    "\n",
    "\n",
    "#head(HOG_level_list)\n",
    "\n",
    "#df %>% rename_at(vars(-Name, -State), ~ paste0(., '_2017'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(HYPOTHESES.a2tea)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HOG_level_list$hypothesis_1\n",
    "HOG_level_list$hypothesis_1 %>%\n",
    "    filter(HOG == \"N0.HOG0000318\")\n",
    "\n",
    "#HOG_level_list$hypothesis_1 %>% group_by(HOG) %>% summarise(sum(total_sigDE))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### creating Venn (or alternative diagrammes for the hypothesis)\n",
    "### -> will be moved to the WebApp and computed there..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create this awesome function to change area of plotting\n",
    "# https://stackoverflow.com/questions/1279003/specify-width-and-height-of-plot\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "# use funtion to set area a bit wider\n",
    "set_plot_dimensions(32, 10)\n",
    "\n",
    "\n",
    "\n",
    "#step 1: removing all species not in hypothesis or keep only those that are\n",
    "# using HOG_level_list and looking at the \"_total\" columns\n",
    "#  only species in this hypothesis are in here ;D\n",
    "# -> still drop all columns not necessary?\n",
    "#step 2: perform check on size of this set\n",
    "# -> if < 5 then create Venn diagramme\n",
    "# -> if > 5 then create alternative plot\n",
    "# create plot and safe as sub element in HYPOTHESES.a2tea object \n",
    "\n",
    "\n",
    "\n",
    "#HOG_DE.a2tea\n",
    "#HOG_level_list[[1]]\n",
    "#choose based on set length 2,3,4,5 after that UpsetR\n",
    "\n",
    "\n",
    "Ortho_intersect_plots <- list()\n",
    "\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "\n",
    "workset <-  HOG_level_list[[i]]\n",
    "# drop columns that are unnecessary\n",
    "workset <- workset %>% select(HOG, ends_with(\"_total\")) \n",
    "# add this point we can easily compute the number of species/ecotypes etc. \n",
    "# #species = (columns of current workset - 1) \n",
    "species_num <- ncol(workset) - 1\n",
    "# change 0 to NAs since we can more easily filter on that\n",
    "is.na(workset) <- workset==0\n",
    "# remove '_total' from column names\n",
    "#str_remove(colnames(workset), \"_total\")\n",
    "workset <- workset %>% setNames(names(workset) %>% stringr::str_replace(\"_total\",\"\"))\n",
    "\n",
    "# for testing purposes - remove Genotype vectors\n",
    "rm(Genotype1,Genotype2,Genotype3)\n",
    "\n",
    "genotypeList <- list()\n",
    "for (j in 1:species_num) {\n",
    "    assign(paste0(\"Genotype\", j), as.vector(simplify_all(workset[!is.na(workset[j+1]),\"HOG\"])[[1]]))\n",
    "    subList <- list(get(paste0(\"Genotype\", j)))\n",
    "    names(subList) <- colnames(workset)[j+1]\n",
    "    genotypeList <- c(genotypeList, subList)\n",
    "}\n",
    "\n",
    "\n",
    "#using UpSetR\n",
    "upsetOrtho <- upset(fromList(genotypeList), \n",
    "                   order.by = \"freq\",\n",
    "                   empty.intersections = \"on\",\n",
    "                   point.size = 3.5, \n",
    "                   line.size = 2, \n",
    "                   mainbar.y.label = \"HOG Intersections\", \n",
    "                   text.scale = c(2, 1.3, 1, 1, 2, 2))\n",
    "\n",
    "p <- upsetOrtho\n",
    "\n",
    "# if we have less than 5 species draw a venn diagramme\n",
    "# combine both plots with some trickery\n",
    "if (species_num < 5) {\n",
    "    \n",
    "# draw venn diagramme\n",
    "orthoVenn <- venn.diagram(genotypeList,\n",
    "                           print.mode = c(\"raw\",\"percent\"),\n",
    "                           sigdigs = 1,\n",
    "                           filename = NULL,\n",
    "                           category = names(genotypeList),\n",
    "                           fill = rainbow(length(names(genotypeList))),\n",
    "                           cat.col = rainbow(length(names(genotypeList))),\n",
    "                           cat.cex = 2,\n",
    "                           cex = 1.5,\n",
    "                           margin = 0.05\n",
    "                           )    \n",
    "    \n",
    "# as.grob from ggplotify is necessary\n",
    "orthoIntersect <- plot_grid(orthoVenn, \n",
    "          as.grob(upsetOrtho), \n",
    "          rel_widths = c(1/3, 2/3))\n",
    "\n",
    "#p <- ggplotify::as.ggplot(orthoIntersect)\n",
    "p <- orthoIntersect\n",
    "    }\n",
    "\n",
    "Ortho_intersect_plots[[i]] <- p\n",
    "\n",
    "}\n",
    "    \n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Ortho_intersect_plots[[1]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### computation of tea value (first idea)\n",
    "\n",
    "expansion ratio (exp/com) (9/4) x expression ratio (com/exp) (2/6) \n",
    "---- don't need (n_species_expanded/n_species_compared); stays equal in a hypothesis; similar to p-values not          comparable between different experiments\n",
    "divided by HOG size (all genes in HOG for this particular hypothesis) x number of significantly regulated genes\n",
    "\n",
    "\n",
    "problem - how to deal with missing values???"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# should I only have hypothesis species (total count and sigDE) HOG_level_list tables?\n",
    "\n",
    "#new column tea-value\n",
    "#if expansion yes continue; else NA value\n",
    "#because we set expansion as criterium, species gene count has to be >=1\n",
    "#sig DE counts can be NA ! - we could just add a pseudocount of 1 at these positions for the calculation\n",
    "\n",
    "#so per line that has expansion = yes:\n",
    "#get all expanded species; sum their gene counts; sum their sigDE counts\n",
    "#get all compared species; sum their gene counts; sum their sigDE counts\n",
    "#expansion_ratio <- divide summed gene count expanded by summed gene count compared\n",
    "#expression_ratio <- divide summed SigDE count expanded by summed SigDE count compared\n",
    "#e2_ratio <- divide expansion_ratio by expression_ratio\n",
    "#tea-value <- divide e2_ratio by total number of genes in HOG and then by total number of SigDE in HOG\n",
    "\n",
    "\n",
    "# outer most loop; going through hypotheses in HOG_level_list\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "    \n",
    "    # access the names of all species important for the hypothesis - expanded_in; compared_to\n",
    "    print(c(unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\")), \n",
    "            unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\"))))\n",
    "    \n",
    "    # creating a relevant species vector might make this more future proof;\n",
    "    # since I might change the completeness of the HOG_level_list elements\n",
    "#    relevant_species <- c(unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\")), \n",
    "#                          unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\")))\n",
    "    \n",
    "    expanded_species <- unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\"))\n",
    "    \n",
    "    compared_species <- unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\"))                              \n",
    "                                  \n",
    "    ## pre-allocate empty vector for tea value with length of nrow of current hypothesis of HOG_level_list\n",
    "    # neat; because NAs already in there so I can just skip if the HOG shows no expansion ;D\n",
    "    tea_value <- rep(NA, nrow(HOG_level_list[[i]]))\n",
    "    \n",
    "    \n",
    "    ## create \"geneCount_workset\" containing HOGs and counts of all relevant species\n",
    "    geneCount_workset <-  HOG_level_list[[i]]\n",
    "\n",
    "    # create geneCount_EXP_workset & geneCount_COMP_workset + drop columns that are unnecessary                  \n",
    "    geneCount_EXP_workset <- geneCount_workset %>% \n",
    "                             select(ends_with(\"_total\")) %>%\n",
    "                             select(contains(expanded_species))\n",
    "                        # also remove the \"_total\" ending\n",
    "    geneCount_EXP_workset <- geneCount_EXP_workset %>% \n",
    "                             setNames(names(geneCount_EXP_workset) %>% \n",
    "                             stringr::str_replace(\"_total\",\"\"))\n",
    "                          \n",
    "    geneCount_COMP_workset <- geneCount_workset %>% \n",
    "                             select(ends_with(\"_total\")) %>%\n",
    "                             select(contains(compared_species))\n",
    "                        # also remove the \"_total\" ending\n",
    "    geneCount_COMP_workset <- geneCount_COMP_workset %>% \n",
    "                             setNames(names(geneCount_COMP_workset) %>% \n",
    "                             stringr::str_replace(\"_total\",\"\"))\n",
    "                    \n",
    "\n",
    "    ## create \"geneSig_workset\" containing HOGs and counts of all relevant species \n",
    "    geneSig_workset <-  HOG_level_list[[i]]\n",
    "    \n",
    "    # create geneSig_EXP_workset & geneSig_COMP_workset + drop columns that are unnecessary                  \n",
    "    geneSig_EXP_workset <- geneSig_workset %>% \n",
    "                             select(ends_with(\"_sigDE\")) %>%\n",
    "                             select(contains(expanded_species))\n",
    "                        # also remove the \"_sigDE\" ending\n",
    "    geneSig_EXP_workset <- geneSig_EXP_workset %>% \n",
    "                             setNames(names(geneSig_EXP_workset) %>% \n",
    "                             stringr::str_replace(\"_sigDE\",\"\"))\n",
    "                          \n",
    "    geneSig_COMP_workset <- geneSig_workset %>% \n",
    "                             select(ends_with(\"_sigDE\")) %>%\n",
    "                             select(contains(compared_species))\n",
    "                        # also remove the \"_sigDE\" ending\n",
    "    geneSig_COMP_workset <- geneSig_COMP_workset %>% \n",
    "                             setNames(names(geneSig_COMP_workset) %>% \n",
    "                             stringr::str_replace(\"_sigDE\",\"\"))\n",
    "\n",
    "                               \n",
    "    # nested loop that goes through current hypothesis HOG_level_list line by line\n",
    "    for (j in 1:nrow(HOG_level_list[[i]])) {\n",
    "        # write value into index slot if current HOG does pass hard filter expansion criterium\n",
    "        # if expansion == \"no\" do nothing; since we prefilled all positions with NA\n",
    "        if (HOG_level_list[[i]][j,\"expansion\"] == \"yes\") {\n",
    "            \n",
    "            ## compute all the different sub-values for the tea_value\n",
    "            # count of all genes in expanded species in HOG\n",
    "            geneCount_EXP_HOG <- geneCount_EXP_workset[j,] %>%\n",
    "                                 dplyr::rowwise() %>%\n",
    "                                 sum()\n",
    "            \n",
    "            # count of all genes in compared species in HOG\n",
    "            geneCount_COMP_HOG <- geneCount_COMP_workset[j,] %>%\n",
    "                                      dplyr::rowwise() %>%\n",
    "                                      sum()\n",
    "            \n",
    "            # sum genes of expanded & compared species in HOG\n",
    "            geneCount_BOTH_HOG <- sum(geneCount_EXP_HOG, geneCount_COMP_HOG)\n",
    "            \n",
    "            \n",
    "            # count of all sig DE genes in expanded species in HOG\n",
    "            geneSig_EXP_HOG <- geneSig_EXP_workset[j,] %>%\n",
    "                                 dplyr::rowwise() %>%\n",
    "                                 sum()\n",
    "            \n",
    "            # count of all sig DE genes in compared species in HOG\n",
    "            geneSig_COMP_HOG <- geneSig_COMP_workset[j,] %>%\n",
    "                                      dplyr::rowwise() %>%\n",
    "                                      sum()\n",
    "            \n",
    "            ## sigDE columns can contain zero (NA in table) counts\n",
    "            ## we exchange this for a pseudocount 1\n",
    "            ## doesn't impact tea_value too much\n",
    "            if (is.na(geneSig_EXP_HOG)) {\n",
    "                geneSig_EXP_HOG <- 1\n",
    "            }\n",
    "            \n",
    "            if (is.na(geneSig_COMP_HOG)) {\n",
    "                geneSig_COMP_HOG <- 1\n",
    "            }\n",
    "            \n",
    "            # sum Sig DE genes of expanded & compared species in HOG\n",
    "            geneSig_BOTH_HOG <- sum(geneSig_EXP_HOG, geneSig_COMP_HOG)\n",
    "\n",
    "            \n",
    "            ## perform tea value calculation\n",
    "            int_tea_value <- (geneCount_EXP_HOG / geneCount_COMP_HOG) * (geneSig_COMP_HOG / geneSig_EXP_HOG) / \n",
    "                                                  (geneCount_BOTH_HOG * geneSig_BOTH_HOG)\n",
    "            \n",
    "            # lastly, add computed value to tea_value at index of current j loop position\n",
    "            tea_value[j] <- int_tea_value\n",
    "        }\n",
    "    }\n",
    "    \n",
    "    ## add tea_value as new column to HOG_level_list table\n",
    "    HOG_level_list[[i]] <- HOG_level_list[[i]] %>% \n",
    "                               add_column(tea_value, .after = \"HOG\")\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "HOG_level_list$hypothesis_2 %>%\n",
    "    arrange(HOG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#c(unlist(str_split(HYPOTHESES.a2tea[[2]]@expanded_in, \";\")), \n",
    "#  unlist(str_split(HYPOTHESES.a2tea[[2]]@compared_to, \";\")))\n",
    "#geneSig_workset <-  HOG_level_list[[1]]\n",
    "#names(geneSig_workset)\n",
    "#head(geneSig_workset %>% setNames(names(geneSig_workset) %>% stringr::str_replace(\"_sigDE\",\"\")))\n",
    "#HOG_level_list[[i]][1,\"expansion\"]\n",
    "#tea_value[2] <- \"asdasdd\"\n",
    "str(tea_value)\n",
    "head(geneCount_EXP_workset[2,])\n",
    "\n",
    "HOG_level_list[[1]][3,c(2,3)] %>%\n",
    "    dplyr::rowwise() %>%\n",
    "    sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(HYPOTHESES.a2tea)\n",
    "#HOG_level_list[[2]] %>% add_column(tea_value, .after = \"HOG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HYPOTHESES.a2tea$hypothesis_2@species_tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### adding the computed cafe p-values to the HOG_level tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for (hypothesis_num in 1:length(HOG_level_list)) {\n",
    "\n",
    "  cafe_tibble <- read_tsv(paste0(\"cafe/\", hypothesis_num, \"/cafe_complete_results/Gamma_family_results.txt\"))[1:2]\n",
    "\n",
    "  HOG_level_list[[hypothesis_num]] <- HOG_level_list[[hypothesis_num]] %>%\n",
    "                                        left_join(., \n",
    "                                                  cafe_tibble, \n",
    "                                                  by = c(\"HOG\" = \"#FamilyID\")\n",
    "                                                 ) %>%\n",
    "                                        rename(cafe_pvalue = pvalue) %>%\n",
    "                                        relocate(cafe_pvalue, .after = tea_value) %>%\n",
    "                                        arrange(tea_value)\n",
    "}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "head(HOG_level_list$hypothesis_1)\n",
    "head(HOG_level_list$hypothesis_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Last step: saving everything to one file which is input for the A2TEA WebApp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(hypotheses, \n",
    "     HYPOTHESES.a2tea, \n",
    "     HOG_DE.a2tea, \n",
    "     HOG_level_list, \n",
    "     Ortho_intersect_plots,\n",
    "     file = \"tea/A2TEA_finished.RData\")\n",
    "\n",
    "# To load the data again\n",
    "#load(\"tea/A2TEA_finished.RData\")\n",
    "\n",
    "#perhaps use this to name output based on time?\n",
    "#paste0(\"tea/A2TEA_finished_\",\n",
    "#    str_replace_all(\n",
    "#    str_replace_all(\n",
    "#        Sys.time(), \":\", \"-\"),\n",
    "#    \" \", \"_\"\n",
    "#                    ), \".RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "final <- load(\"tea/A2TEA_finished.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 20913 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>HOG</th><th scope=col>tea_value</th><th scope=col>cafe_pvalue</th><th scope=col>Hordeum_vulgare_total</th><th scope=col>Zea_mays_total</th><th scope=col>expansion</th><th scope=col>Arabidopsis_thaliana_sigDE</th><th scope=col>Hordeum_vulgare_sigDE</th><th scope=col>Zea_mays_sigDE</th><th scope=col>total_sigDE</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>N0.HOG0000047</td><td>0.0009032634</td><td>0.000</td><td> 31</td><td>8</td><td>yes</td><td>NA</td><td>10</td><td>NA</td><td>10</td></tr>\n",
       "\t<tr><td>N0.HOG0000063</td><td>0.0022522523</td><td>0.000</td><td> 35</td><td>2</td><td>yes</td><td>NA</td><td>14</td><td>NA</td><td>14</td></tr>\n",
       "\t<tr><td>N0.HOG0000044</td><td>0.0023530762</td><td>0.000</td><td> 41</td><td>3</td><td>yes</td><td>NA</td><td>11</td><td>NA</td><td>11</td></tr>\n",
       "\t<tr><td>N0.HOG0000049</td><td>0.0025067751</td><td>0.000</td><td> 37</td><td>4</td><td>yes</td><td>NA</td><td> 9</td><td>NA</td><td> 9</td></tr>\n",
       "\t<tr><td>N0.HOG0000297</td><td>0.0033482143</td><td>0.008</td><td> 12</td><td>4</td><td>yes</td><td>NA</td><td> 7</td><td>NA</td><td> 7</td></tr>\n",
       "\t<tr><td>N0.HOG0000073</td><td>0.0033769063</td><td>0.000</td><td> 31</td><td>3</td><td>yes</td><td>NA</td><td> 9</td><td>NA</td><td> 9</td></tr>\n",
       "\t<tr><td>N0.HOG0000072</td><td>0.0046218487</td><td>0.000</td><td> 33</td><td>1</td><td>yes</td><td>NA</td><td>14</td><td>NA</td><td>14</td></tr>\n",
       "\t<tr><td>N0.HOG0000655</td><td>0.0046296296</td><td>0.003</td><td> 10</td><td>2</td><td>yes</td><td>NA</td><td> 9</td><td>NA</td><td> 9</td></tr>\n",
       "\t<tr><td>N0.HOG0000182</td><td>0.0050595238</td><td>0.000</td><td> 17</td><td>3</td><td>yes</td><td>NA</td><td> 7</td><td>NA</td><td> 7</td></tr>\n",
       "\t<tr><td>N0.HOG0000294</td><td>0.0060763889</td><td>0.000</td><td> 14</td><td>2</td><td>yes</td><td>NA</td><td> 8</td><td>NA</td><td> 8</td></tr>\n",
       "\t<tr><td>N0.HOG0000078</td><td>0.0062160062</td><td>0.000</td><td> 32</td><td>1</td><td>yes</td><td>NA</td><td>12</td><td>NA</td><td>12</td></tr>\n",
       "\t<tr><td>N0.HOG0000110</td><td>0.0064300412</td><td>0.000</td><td> 25</td><td>2</td><td>yes</td><td>NA</td><td> 8</td><td>NA</td><td> 8</td></tr>\n",
       "\t<tr><td>N0.HOG0000530</td><td>0.0069930070</td><td>0.000</td><td> 12</td><td>1</td><td>yes</td><td>NA</td><td>11</td><td>NA</td><td>11</td></tr>\n",
       "\t<tr><td>N0.HOG0000455</td><td>0.0089285714</td><td>0.095</td><td>  6</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000183</td><td>0.0094444444</td><td>0.000</td><td> 17</td><td>3</td><td>yes</td><td>NA</td><td> 5</td><td>NA</td><td> 5</td></tr>\n",
       "\t<tr><td>N0.HOG0000489</td><td>0.0095238095</td><td>0.013</td><td>  8</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000785</td><td>0.0097402597</td><td>0.004</td><td>  9</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000624</td><td>0.0100000000</td><td>0.001</td><td>  9</td><td>1</td><td>yes</td><td>NA</td><td> 9</td><td> 1</td><td>10</td></tr>\n",
       "\t<tr><td>N0.HOG0000272</td><td>0.0100732601</td><td>0.001</td><td> 11</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000148</td><td>0.0102272727</td><td>0.000</td><td> 18</td><td>4</td><td>yes</td><td>NA</td><td> 4</td><td>NA</td><td> 4</td></tr>\n",
       "\t<tr><td>N0.HOG0000108</td><td>0.0108225108</td><td>0.000</td><td> 20</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000128</td><td>0.0109126984</td><td>0.000</td><td> 22</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000083</td><td>0.0111111111</td><td>0.000</td><td> 28</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000001</td><td>0.0118016904</td><td>0.000</td><td>229</td><td>2</td><td>yes</td><td>NA</td><td> 6</td><td>NA</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000527</td><td>0.0128205128</td><td>0.010</td><td> 10</td><td>3</td><td>yes</td><td>NA</td><td> 4</td><td>NA</td><td> 4</td></tr>\n",
       "\t<tr><td>N0.HOG0001014</td><td>0.0129629630</td><td>0.027</td><td>  7</td><td>2</td><td>yes</td><td>NA</td><td> 5</td><td> 1</td><td> 6</td></tr>\n",
       "\t<tr><td>N0.HOG0000293</td><td>0.0130208333</td><td>0.000</td><td> 15</td><td>1</td><td>yes</td><td>NA</td><td> 8</td><td>NA</td><td> 8</td></tr>\n",
       "\t<tr><td>N0.HOG0000606</td><td>0.0133333333</td><td>0.013</td><td>  8</td><td>2</td><td>yes</td><td>NA</td><td> 5</td><td>NA</td><td> 5</td></tr>\n",
       "\t<tr><td>N0.HOG0000714</td><td>0.0133333333</td><td>0.013</td><td>  8</td><td>2</td><td>yes</td><td>NA</td><td> 5</td><td>NA</td><td> 5</td></tr>\n",
       "\t<tr><td>N0.HOG0001059</td><td>0.0133333333</td><td>0.013</td><td>  8</td><td>2</td><td>yes</td><td>NA</td><td> 5</td><td> 1</td><td> 6</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>N0.HOG0020883</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020884</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020885</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020886</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020887</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020888</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020889</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020890</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020891</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020892</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020893</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020894</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020895</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020896</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020897</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020898</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020899</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020900</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020901</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020902</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020903</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020904</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020905</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020906</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020907</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020908</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020909</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020910</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020911</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N0.HOG0020912</td><td>NA</td><td>NA</td><td>0</td><td>2</td><td>no</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 20913 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " HOG & tea\\_value & cafe\\_pvalue & Hordeum\\_vulgare\\_total & Zea\\_mays\\_total & expansion & Arabidopsis\\_thaliana\\_sigDE & Hordeum\\_vulgare\\_sigDE & Zea\\_mays\\_sigDE & total\\_sigDE\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t N0.HOG0000047 & 0.0009032634 & 0.000 &  31 & 8 & yes & NA & 10 & NA & 10\\\\\n",
       "\t N0.HOG0000063 & 0.0022522523 & 0.000 &  35 & 2 & yes & NA & 14 & NA & 14\\\\\n",
       "\t N0.HOG0000044 & 0.0023530762 & 0.000 &  41 & 3 & yes & NA & 11 & NA & 11\\\\\n",
       "\t N0.HOG0000049 & 0.0025067751 & 0.000 &  37 & 4 & yes & NA &  9 & NA &  9\\\\\n",
       "\t N0.HOG0000297 & 0.0033482143 & 0.008 &  12 & 4 & yes & NA &  7 & NA &  7\\\\\n",
       "\t N0.HOG0000073 & 0.0033769063 & 0.000 &  31 & 3 & yes & NA &  9 & NA &  9\\\\\n",
       "\t N0.HOG0000072 & 0.0046218487 & 0.000 &  33 & 1 & yes & NA & 14 & NA & 14\\\\\n",
       "\t N0.HOG0000655 & 0.0046296296 & 0.003 &  10 & 2 & yes & NA &  9 & NA &  9\\\\\n",
       "\t N0.HOG0000182 & 0.0050595238 & 0.000 &  17 & 3 & yes & NA &  7 & NA &  7\\\\\n",
       "\t N0.HOG0000294 & 0.0060763889 & 0.000 &  14 & 2 & yes & NA &  8 & NA &  8\\\\\n",
       "\t N0.HOG0000078 & 0.0062160062 & 0.000 &  32 & 1 & yes & NA & 12 & NA & 12\\\\\n",
       "\t N0.HOG0000110 & 0.0064300412 & 0.000 &  25 & 2 & yes & NA &  8 & NA &  8\\\\\n",
       "\t N0.HOG0000530 & 0.0069930070 & 0.000 &  12 & 1 & yes & NA & 11 & NA & 11\\\\\n",
       "\t N0.HOG0000455 & 0.0089285714 & 0.095 &   6 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000183 & 0.0094444444 & 0.000 &  17 & 3 & yes & NA &  5 & NA &  5\\\\\n",
       "\t N0.HOG0000489 & 0.0095238095 & 0.013 &   8 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000785 & 0.0097402597 & 0.004 &   9 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000624 & 0.0100000000 & 0.001 &   9 & 1 & yes & NA &  9 &  1 & 10\\\\\n",
       "\t N0.HOG0000272 & 0.0100732601 & 0.001 &  11 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000148 & 0.0102272727 & 0.000 &  18 & 4 & yes & NA &  4 & NA &  4\\\\\n",
       "\t N0.HOG0000108 & 0.0108225108 & 0.000 &  20 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000128 & 0.0109126984 & 0.000 &  22 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000083 & 0.0111111111 & 0.000 &  28 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000001 & 0.0118016904 & 0.000 & 229 & 2 & yes & NA &  6 & NA &  6\\\\\n",
       "\t N0.HOG0000527 & 0.0128205128 & 0.010 &  10 & 3 & yes & NA &  4 & NA &  4\\\\\n",
       "\t N0.HOG0001014 & 0.0129629630 & 0.027 &   7 & 2 & yes & NA &  5 &  1 &  6\\\\\n",
       "\t N0.HOG0000293 & 0.0130208333 & 0.000 &  15 & 1 & yes & NA &  8 & NA &  8\\\\\n",
       "\t N0.HOG0000606 & 0.0133333333 & 0.013 &   8 & 2 & yes & NA &  5 & NA &  5\\\\\n",
       "\t N0.HOG0000714 & 0.0133333333 & 0.013 &   8 & 2 & yes & NA &  5 & NA &  5\\\\\n",
       "\t N0.HOG0001059 & 0.0133333333 & 0.013 &   8 & 2 & yes & NA &  5 &  1 &  6\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t N0.HOG0020883 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020884 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020885 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020886 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020887 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020888 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020889 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020890 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020891 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020892 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020893 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020894 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020895 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020896 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020897 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020898 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020899 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020900 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020901 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020902 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020903 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020904 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020905 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020906 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020907 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020908 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020909 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020910 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020911 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\t N0.HOG0020912 & NA & NA & 0 & 2 & no & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 20913 × 10\n",
       "\n",
       "| HOG &lt;chr&gt; | tea_value &lt;dbl&gt; | cafe_pvalue &lt;dbl&gt; | Hordeum_vulgare_total &lt;dbl&gt; | Zea_mays_total &lt;dbl&gt; | expansion &lt;chr&gt; | Arabidopsis_thaliana_sigDE &lt;dbl&gt; | Hordeum_vulgare_sigDE &lt;dbl&gt; | Zea_mays_sigDE &lt;dbl&gt; | total_sigDE &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| N0.HOG0000047 | 0.0009032634 | 0.000 |  31 | 8 | yes | NA | 10 | NA | 10 |\n",
       "| N0.HOG0000063 | 0.0022522523 | 0.000 |  35 | 2 | yes | NA | 14 | NA | 14 |\n",
       "| N0.HOG0000044 | 0.0023530762 | 0.000 |  41 | 3 | yes | NA | 11 | NA | 11 |\n",
       "| N0.HOG0000049 | 0.0025067751 | 0.000 |  37 | 4 | yes | NA |  9 | NA |  9 |\n",
       "| N0.HOG0000297 | 0.0033482143 | 0.008 |  12 | 4 | yes | NA |  7 | NA |  7 |\n",
       "| N0.HOG0000073 | 0.0033769063 | 0.000 |  31 | 3 | yes | NA |  9 | NA |  9 |\n",
       "| N0.HOG0000072 | 0.0046218487 | 0.000 |  33 | 1 | yes | NA | 14 | NA | 14 |\n",
       "| N0.HOG0000655 | 0.0046296296 | 0.003 |  10 | 2 | yes | NA |  9 | NA |  9 |\n",
       "| N0.HOG0000182 | 0.0050595238 | 0.000 |  17 | 3 | yes | NA |  7 | NA |  7 |\n",
       "| N0.HOG0000294 | 0.0060763889 | 0.000 |  14 | 2 | yes | NA |  8 | NA |  8 |\n",
       "| N0.HOG0000078 | 0.0062160062 | 0.000 |  32 | 1 | yes | NA | 12 | NA | 12 |\n",
       "| N0.HOG0000110 | 0.0064300412 | 0.000 |  25 | 2 | yes | NA |  8 | NA |  8 |\n",
       "| N0.HOG0000530 | 0.0069930070 | 0.000 |  12 | 1 | yes | NA | 11 | NA | 11 |\n",
       "| N0.HOG0000455 | 0.0089285714 | 0.095 |   6 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000183 | 0.0094444444 | 0.000 |  17 | 3 | yes | NA |  5 | NA |  5 |\n",
       "| N0.HOG0000489 | 0.0095238095 | 0.013 |   8 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000785 | 0.0097402597 | 0.004 |   9 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000624 | 0.0100000000 | 0.001 |   9 | 1 | yes | NA |  9 |  1 | 10 |\n",
       "| N0.HOG0000272 | 0.0100732601 | 0.001 |  11 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000148 | 0.0102272727 | 0.000 |  18 | 4 | yes | NA |  4 | NA |  4 |\n",
       "| N0.HOG0000108 | 0.0108225108 | 0.000 |  20 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000128 | 0.0109126984 | 0.000 |  22 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000083 | 0.0111111111 | 0.000 |  28 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000001 | 0.0118016904 | 0.000 | 229 | 2 | yes | NA |  6 | NA |  6 |\n",
       "| N0.HOG0000527 | 0.0128205128 | 0.010 |  10 | 3 | yes | NA |  4 | NA |  4 |\n",
       "| N0.HOG0001014 | 0.0129629630 | 0.027 |   7 | 2 | yes | NA |  5 |  1 |  6 |\n",
       "| N0.HOG0000293 | 0.0130208333 | 0.000 |  15 | 1 | yes | NA |  8 | NA |  8 |\n",
       "| N0.HOG0000606 | 0.0133333333 | 0.013 |   8 | 2 | yes | NA |  5 | NA |  5 |\n",
       "| N0.HOG0000714 | 0.0133333333 | 0.013 |   8 | 2 | yes | NA |  5 | NA |  5 |\n",
       "| N0.HOG0001059 | 0.0133333333 | 0.013 |   8 | 2 | yes | NA |  5 |  1 |  6 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| N0.HOG0020883 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020884 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020885 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020886 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020887 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020888 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020889 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020890 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020891 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020892 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020893 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020894 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020895 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020896 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020897 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020898 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020899 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020900 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020901 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020902 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020903 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020904 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020905 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020906 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020907 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020908 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020909 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020910 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020911 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "| N0.HOG0020912 | NA | NA | 0 | 2 | no | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "      HOG           tea_value    cafe_pvalue Hordeum_vulgare_total\n",
       "1     N0.HOG0000047 0.0009032634 0.000        31                  \n",
       "2     N0.HOG0000063 0.0022522523 0.000        35                  \n",
       "3     N0.HOG0000044 0.0023530762 0.000        41                  \n",
       "4     N0.HOG0000049 0.0025067751 0.000        37                  \n",
       "5     N0.HOG0000297 0.0033482143 0.008        12                  \n",
       "6     N0.HOG0000073 0.0033769063 0.000        31                  \n",
       "7     N0.HOG0000072 0.0046218487 0.000        33                  \n",
       "8     N0.HOG0000655 0.0046296296 0.003        10                  \n",
       "9     N0.HOG0000182 0.0050595238 0.000        17                  \n",
       "10    N0.HOG0000294 0.0060763889 0.000        14                  \n",
       "11    N0.HOG0000078 0.0062160062 0.000        32                  \n",
       "12    N0.HOG0000110 0.0064300412 0.000        25                  \n",
       "13    N0.HOG0000530 0.0069930070 0.000        12                  \n",
       "14    N0.HOG0000455 0.0089285714 0.095         6                  \n",
       "15    N0.HOG0000183 0.0094444444 0.000        17                  \n",
       "16    N0.HOG0000489 0.0095238095 0.013         8                  \n",
       "17    N0.HOG0000785 0.0097402597 0.004         9                  \n",
       "18    N0.HOG0000624 0.0100000000 0.001         9                  \n",
       "19    N0.HOG0000272 0.0100732601 0.001        11                  \n",
       "20    N0.HOG0000148 0.0102272727 0.000        18                  \n",
       "21    N0.HOG0000108 0.0108225108 0.000        20                  \n",
       "22    N0.HOG0000128 0.0109126984 0.000        22                  \n",
       "23    N0.HOG0000083 0.0111111111 0.000        28                  \n",
       "24    N0.HOG0000001 0.0118016904 0.000       229                  \n",
       "25    N0.HOG0000527 0.0128205128 0.010        10                  \n",
       "26    N0.HOG0001014 0.0129629630 0.027         7                  \n",
       "27    N0.HOG0000293 0.0130208333 0.000        15                  \n",
       "28    N0.HOG0000606 0.0133333333 0.013         8                  \n",
       "29    N0.HOG0000714 0.0133333333 0.013         8                  \n",
       "30    N0.HOG0001059 0.0133333333 0.013         8                  \n",
       "⋮     ⋮             ⋮            ⋮           ⋮                    \n",
       "20884 N0.HOG0020883 NA           NA          0                    \n",
       "20885 N0.HOG0020884 NA           NA          0                    \n",
       "20886 N0.HOG0020885 NA           NA          0                    \n",
       "20887 N0.HOG0020886 NA           NA          0                    \n",
       "20888 N0.HOG0020887 NA           NA          0                    \n",
       "20889 N0.HOG0020888 NA           NA          0                    \n",
       "20890 N0.HOG0020889 NA           NA          0                    \n",
       "20891 N0.HOG0020890 NA           NA          0                    \n",
       "20892 N0.HOG0020891 NA           NA          0                    \n",
       "20893 N0.HOG0020892 NA           NA          0                    \n",
       "20894 N0.HOG0020893 NA           NA          0                    \n",
       "20895 N0.HOG0020894 NA           NA          0                    \n",
       "20896 N0.HOG0020895 NA           NA          0                    \n",
       "20897 N0.HOG0020896 NA           NA          0                    \n",
       "20898 N0.HOG0020897 NA           NA          0                    \n",
       "20899 N0.HOG0020898 NA           NA          0                    \n",
       "20900 N0.HOG0020899 NA           NA          0                    \n",
       "20901 N0.HOG0020900 NA           NA          0                    \n",
       "20902 N0.HOG0020901 NA           NA          0                    \n",
       "20903 N0.HOG0020902 NA           NA          0                    \n",
       "20904 N0.HOG0020903 NA           NA          0                    \n",
       "20905 N0.HOG0020904 NA           NA          0                    \n",
       "20906 N0.HOG0020905 NA           NA          0                    \n",
       "20907 N0.HOG0020906 NA           NA          0                    \n",
       "20908 N0.HOG0020907 NA           NA          0                    \n",
       "20909 N0.HOG0020908 NA           NA          0                    \n",
       "20910 N0.HOG0020909 NA           NA          0                    \n",
       "20911 N0.HOG0020910 NA           NA          0                    \n",
       "20912 N0.HOG0020911 NA           NA          0                    \n",
       "20913 N0.HOG0020912 NA           NA          0                    \n",
       "      Zea_mays_total expansion Arabidopsis_thaliana_sigDE Hordeum_vulgare_sigDE\n",
       "1     8              yes       NA                         10                   \n",
       "2     2              yes       NA                         14                   \n",
       "3     3              yes       NA                         11                   \n",
       "4     4              yes       NA                          9                   \n",
       "5     4              yes       NA                          7                   \n",
       "6     3              yes       NA                          9                   \n",
       "7     1              yes       NA                         14                   \n",
       "8     2              yes       NA                          9                   \n",
       "9     3              yes       NA                          7                   \n",
       "10    2              yes       NA                          8                   \n",
       "11    1              yes       NA                         12                   \n",
       "12    2              yes       NA                          8                   \n",
       "13    1              yes       NA                         11                   \n",
       "14    2              yes       NA                          6                   \n",
       "15    3              yes       NA                          5                   \n",
       "16    2              yes       NA                          6                   \n",
       "17    2              yes       NA                          6                   \n",
       "18    1              yes       NA                          9                   \n",
       "19    2              yes       NA                          6                   \n",
       "20    4              yes       NA                          4                   \n",
       "21    2              yes       NA                          6                   \n",
       "22    2              yes       NA                          6                   \n",
       "23    2              yes       NA                          6                   \n",
       "24    2              yes       NA                          6                   \n",
       "25    3              yes       NA                          4                   \n",
       "26    2              yes       NA                          5                   \n",
       "27    1              yes       NA                          8                   \n",
       "28    2              yes       NA                          5                   \n",
       "29    2              yes       NA                          5                   \n",
       "30    2              yes       NA                          5                   \n",
       "⋮     ⋮              ⋮         ⋮                          ⋮                    \n",
       "20884 2              no        NA                         NA                   \n",
       "20885 2              no        NA                         NA                   \n",
       "20886 2              no        NA                         NA                   \n",
       "20887 2              no        NA                         NA                   \n",
       "20888 2              no        NA                         NA                   \n",
       "20889 2              no        NA                         NA                   \n",
       "20890 2              no        NA                         NA                   \n",
       "20891 2              no        NA                         NA                   \n",
       "20892 2              no        NA                         NA                   \n",
       "20893 2              no        NA                         NA                   \n",
       "20894 2              no        NA                         NA                   \n",
       "20895 2              no        NA                         NA                   \n",
       "20896 2              no        NA                         NA                   \n",
       "20897 2              no        NA                         NA                   \n",
       "20898 2              no        NA                         NA                   \n",
       "20899 2              no        NA                         NA                   \n",
       "20900 2              no        NA                         NA                   \n",
       "20901 2              no        NA                         NA                   \n",
       "20902 2              no        NA                         NA                   \n",
       "20903 2              no        NA                         NA                   \n",
       "20904 2              no        NA                         NA                   \n",
       "20905 2              no        NA                         NA                   \n",
       "20906 2              no        NA                         NA                   \n",
       "20907 2              no        NA                         NA                   \n",
       "20908 2              no        NA                         NA                   \n",
       "20909 2              no        NA                         NA                   \n",
       "20910 2              no        NA                         NA                   \n",
       "20911 2              no        NA                         NA                   \n",
       "20912 2              no        NA                         NA                   \n",
       "20913 2              no        NA                         NA                   \n",
       "      Zea_mays_sigDE total_sigDE\n",
       "1     NA             10         \n",
       "2     NA             14         \n",
       "3     NA             11         \n",
       "4     NA              9         \n",
       "5     NA              7         \n",
       "6     NA              9         \n",
       "7     NA             14         \n",
       "8     NA              9         \n",
       "9     NA              7         \n",
       "10    NA              8         \n",
       "11    NA             12         \n",
       "12    NA              8         \n",
       "13    NA             11         \n",
       "14    NA              6         \n",
       "15    NA              5         \n",
       "16    NA              6         \n",
       "17    NA              6         \n",
       "18     1             10         \n",
       "19    NA              6         \n",
       "20    NA              4         \n",
       "21    NA              6         \n",
       "22    NA              6         \n",
       "23    NA              6         \n",
       "24    NA              6         \n",
       "25    NA              4         \n",
       "26     1              6         \n",
       "27    NA              8         \n",
       "28    NA              5         \n",
       "29    NA              5         \n",
       "30     1              6         \n",
       "⋮     ⋮              ⋮          \n",
       "20884 NA             NA         \n",
       "20885 NA             NA         \n",
       "20886 NA             NA         \n",
       "20887 NA             NA         \n",
       "20888 NA             NA         \n",
       "20889 NA             NA         \n",
       "20890 NA             NA         \n",
       "20891 NA             NA         \n",
       "20892 NA             NA         \n",
       "20893 NA             NA         \n",
       "20894 NA             NA         \n",
       "20895 NA             NA         \n",
       "20896 NA             NA         \n",
       "20897 NA             NA         \n",
       "20898 NA             NA         \n",
       "20899 NA             NA         \n",
       "20900 NA             NA         \n",
       "20901 NA             NA         \n",
       "20902 NA             NA         \n",
       "20903 NA             NA         \n",
       "20904 NA             NA         \n",
       "20905 NA             NA         \n",
       "20906 NA             NA         \n",
       "20907 NA             NA         \n",
       "20908 NA             NA         \n",
       "20909 NA             NA         \n",
       "20910 NA             NA         \n",
       "20911 NA             NA         \n",
       "20912 NA             NA         \n",
       "20913 NA             NA         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "HOG_level_list$hypothesis_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "HYPOTHESES.a2tea$hypothesis_1@expanded_OGs$N0.HOG0000207@blast_table"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
