{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combining all downstream used A2TEA data into one RDS object\n",
    "## Also performing necessary data manipulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "###packages currently installed via conda\n",
    "\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(version = \"3.12\")\n",
    "\n",
    "#BiocManager::install(\"DESeq2\")\n",
    "#BiocManager::install(\"ggtree\")\n",
    "#BiocManager::install(\"Biostrings\")\n",
    "\n",
    "#lfcShrinkage isn't installable at the moment - problem for later\n",
    "#install.packages(\"mvtnorm\")\n",
    "#BiocManager::install(\"apeglm\")\n",
    "\n",
    "#install.packages(\"data.table\")\n",
    "\n",
    "#install.packages(\"seqinr\")\n",
    "#install.packages(\"VennDiagram\")\n",
    "#install.packages(\"UpSetR\")\n",
    "#install.packages(\"cowplot\")\n",
    "#install.packages(\"ggplotify\")\n",
    "\n",
    "###############################################\n",
    "\n",
    "\n",
    "if (!requireNamespace(\"BiocManager\", quietly=TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install(\"ggtree\")\n",
    "BiocManager::install(\"ggtreeExtra\")\n",
    "BiocManager::install(\"Biostrings\")\n",
    "\n",
    "install.packages(\"VennDiagram\")\n",
    "install.packages(\"UpSetR\")\n",
    "install.packages(\"cowplot\")\n",
    "install.packages(\"ggplotify\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘DESeq2’ was built under R version 4.0.3”\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Warning message:\n",
      "“package ‘S4Vectors’ was built under R version 4.0.3”\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘IRanges’ was built under R version 4.0.3”\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomicRanges’ was built under R version 4.0.3”\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomeInfoDb’ was built under R version 4.0.3”\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Warning message:\n",
      "“package ‘SummarizedExperiment’ was built under R version 4.0.3”\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘MatrixGenerics’ was built under R version 4.0.3”\n",
      "Loading required package: matrixStats\n",
      "\n",
      "Warning message:\n",
      "“package ‘matrixStats’ was built under R version 4.0.3”\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Warning message:\n",
      "“package ‘Biobase’ was built under R version 4.0.3”\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────── tidyverse 1.3.0 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.3     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.0     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.5\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "Warning message:\n",
      "“package ‘ggplot2’ was built under R version 4.0.3”\n",
      "Warning message:\n",
      "“package ‘tibble’ was built under R version 4.0.3”\n",
      "Warning message:\n",
      "“package ‘tidyr’ was built under R version 4.0.3”\n",
      "Warning message:\n",
      "“package ‘readr’ was built under R version 4.0.3”\n",
      "Warning message:\n",
      "“package ‘dplyr’ was built under R version 4.0.3”\n",
      "Warning message:\n",
      "“package ‘forcats’ was built under R version 4.0.3”\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcount()\u001b[39m      masks \u001b[34mmatrixStats\u001b[39m::count()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "Registered S3 method overwritten by 'treeio':\n",
      "  method     from\n",
      "  root.phylo ape \n",
      "\n",
      "ggtree v2.4.1  For help: https://yulab-smu.top/treedata-book/\n",
      "\n",
      "If you use ggtree in published research, please cite the most appropriate paper(s):\n",
      "\n",
      "\u001b[36m-\u001b[39m Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics, 2020, 69:e96. doi:10.1002/cpbi.96\n",
      "\u001b[36m-\u001b[39m Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194\n",
      "\u001b[36m-\u001b[39m Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36. doi:10.1111/2041-210X.12628\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ggtree’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "Warning message:\n",
      "“package ‘XVector’ was built under R version 4.0.3”\n",
      "\n",
      "Attaching package: ‘XVector’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    compact\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘seqinr’ was built under R version 4.0.3”\n",
      "\n",
      "Attaching package: ‘seqinr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biostrings’:\n",
      "\n",
      "    translate\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "Loading required package: futile.logger\n",
      "\n",
      "\n",
      "Attaching package: ‘VennDiagram’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ggtree’:\n",
      "\n",
      "    rotate\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ape’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:VennDiagram’:\n",
      "\n",
      "    rotate\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:seqinr’:\n",
      "\n",
      "    as.alignment, consensus\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biostrings’:\n",
      "\n",
      "    complement\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ggtree’:\n",
      "\n",
      "    rotate\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(DESeq2)\n",
    "#library(BiocGenerics) # comes with DESeq2\n",
    "#library(apeglm) # currently difficulties with install # better DESseq2 calc; later \n",
    "\n",
    "library(tidyverse)\n",
    "library(ggtree)\n",
    "library(Biostrings)\n",
    "library(seqinr)\n",
    "library(VennDiagram)\n",
    "library(UpSetR)\n",
    "library(cowplot)\n",
    "library(ggplotify)\n",
    "library(ape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Find all DESeq2 differential expression RDS objects and load them\n",
    "##### Names are given based on the species/ecotype/etc. name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dea_list <- list.files(path = \"R/deseq2/dea_final/\", pattern = \"dea_*\", full.names=TRUE)\n",
    "dea_list_short <- list.files(path = \"R/deseq2/dea_final/\", pattern = \"dea_*\", full.names=FALSE)\n",
    "\n",
    "listRDS <- lapply(dea_list, readRDS, .GlobalEnv)\n",
    "\n",
    "for (i in 1:length(listRDS)){\n",
    "    assign(str_sub(dea_list_short[[i]], start=5), listRDS[[i]])\n",
    "}\n",
    "\n",
    "# remove unused combined list object\n",
    "rm(listRDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#nice for later - but needs install of package \"apeglm\"\n",
    "\n",
    "#link on logFC shrinking\n",
    "#https://support.bioconductor.org/p/77461/\n",
    "#?lfcShrink\n",
    "#lfcShrink(arabidopsis_dea, coef=2, type=\"apeglm\")\n",
    "#resultsNames(arabidopsis_dea)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DESeq2 testing\n",
    "#At <- readRDS(\"R/deseq2/dea_final/dea_Arabidopsis_thaliana\")\n",
    "#DESeq2::results(At)\n",
    "#as.data.frame(results(Arabidopsis_thaliana))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_list <- vector()\n",
    "\n",
    "for (i in 1:length(dea_list_short)){\n",
    "    species_list <- c(species_list, str_sub(dea_list_short[[i]], start=5))\n",
    "}\n",
    "\n",
    "list_AllSpeciesDEResultsDataFrames <- list()\n",
    "\n",
    "for (i in species_list) {\n",
    "    # individual species file could be deleted but perhaps I'll need them later; kept for now...\n",
    "    assign(paste0(i, \"_DEresultsTable\"), as.data.frame(results(get(i))))\n",
    "    # create gene name vector here (when list is constructed rownames become unique species-gene combination)\n",
    "    gene <- rownames(get(paste0(i, \"_DEresultsTable\")))\n",
    "    df_with_genes <- add_column(get(paste0(i, \"_DEresultsTable\")), gene, .before = \"baseMean\")\n",
    "    # also add a species column which will will also come in handy during the shiny steps\n",
    "    species <- replicate(nrow(df_with_genes), i)\n",
    "    df_with_genes <- add_column(df_with_genes, species, .before = \"gene\")\n",
    "    # create list of dataframes, which will come in handy\n",
    "    list_AllSpeciesDEResultsDataFrames[[i]] <- df_with_genes\n",
    "    # list_AllSpeciesDEResultsDataFrames[[i]] <- (get(paste0(i, \"_DEresultsTable\")))\n",
    "}\n",
    "\n",
    "#str(list_AllSpeciesDEResultsDataFrames)\n",
    "\n",
    "#create combined \"mega\" dataframe of all species, which is going  to be used for shiny lookup\n",
    "\n",
    "combined_AllSpeciesDEResultsDataFrames <- do.call(\"rbind\", list_AllSpeciesDEResultsDataFrames)\n",
    "\n",
    "\n",
    "#str(combined_AllSpeciesDEResultsDataFrames)\n",
    "#head(combined_AllSpeciesDEResultsDataFrames)\n",
    "#head(list_AllSpeciesDEResultsDataFrames[[1]])\n",
    "#?rbind"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### creating long format HOG-genes relation table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HOG_file_raw <- read_delim(\"orthofinder/final-results/Phylogenetic_Hierarchical_Orthogroups/N0.tsv\", delim = \"\\t\")\n",
    "\n",
    "#drop the OG, Gene Tree Parent Clade\n",
    "HOG_file_short <- HOG_file_raw %>% select (-c(OG, `Gene Tree Parent Clade`)) \n",
    "\n",
    "\n",
    "#merge all species columns into one and remove the solo ones\n",
    "#use all colnames except first - thus all species\n",
    "HOG_file_merged <- HOG_file_short %>% unite(x,\n",
    "                                          c(colnames(HOG_file_short)[-1]),\n",
    "                                          sep = \",\", \n",
    "                                          na.rm = TRUE,\n",
    "                                          remove = TRUE)\n",
    "\n",
    "#transform into long format - each row 1 gene and it's corresponding HOG in seperate columns\n",
    "HOG_file_long <- HOG_file_merged %>%\n",
    "                    mutate(unpacked = str_split(x, \",\")) %>%\n",
    "                    unnest(cols = c(unpacked)) %>%\n",
    "                    mutate(genes = str_trim(unpacked)) %>% \n",
    "                    select(-c(x, unpacked)) %>% \n",
    "                    rename(gene = genes)\n",
    "\n",
    "str(HOG_file_long)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### adding specific HOG or singleton info as HOG column to DE tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#?full_join\n",
    "HOG_DE.a2tea <- full_join(combined_AllSpeciesDEResultsDataFrames, HOG_file_long, \n",
    "              by = c(\"gene\")) %>% \n",
    "                replace_na(list(HOG = \"singleton\"))\n",
    "\n",
    "#add column for significance - default level is padj <= 0.1\n",
    "\n",
    "#subset tidyverse with the right functions ;D\n",
    "#filter(HOG_DE.a2tea, padj < 0.05)\n",
    "#?drop_na\n",
    "\n",
    "significant <- vector()\n",
    "for (FDR in HOG_DE.a2tea$padj) {\n",
    "    if (!is.na(FDR) && FDR < 0.1) {\n",
    "        significant <- c(significant, \"yes\")\n",
    "    }\n",
    "    else if (!is.na(FDR) && FDR > 0.1) {\n",
    "        significant <- c(significant, \"no\")\n",
    "    }\n",
    "    else if (is.na(FDR)) {\n",
    "        significant <- c(significant, \"no\")\n",
    "    }\n",
    "}\n",
    "\n",
    "HOG_DE.a2tea <- add_column(HOG_DE.a2tea, significant, .after = \"padj\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#?full_join"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load in the rest of the data - hypotheses, trees, fasta and msa for the start \n",
    "### + toDo general stats, especially once Orthofinder calculates for HOG)\n",
    "### +toDo Venn DIagrammes (until 5 species or so then alternative plots)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#extended_BLAST_hits <- readRDS(\"tea/2/extended_BLAST_hits/extended_BLAST_hits.RDS\")\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\",\n",
    "                    genes_extend_hits=\"tbl_df\")\n",
    "         )\n",
    "#expanded_OGs_short\n",
    "#extended_BLAST_hits[[\"N0.HOG0000207\"]]@nrow_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AAStringSet object of length 55:\n",
       "     width seq                                              names               \n",
       " [1]   772 -----------------------...GIHFVDHEAVL----------- HORVU2Hr1G007260\n",
       " [2]   772 -----------------------...GIHVVP---------------- HORVU2Hr1G007230\n",
       " [3]   772 -----------------------...VNVVPV---------------- Zm00001d004305\n",
       " [4]   772 -----------------------...IHLVPI---------------- Zm00001d004306\n",
       " [5]   772 -----------------------...INLVPI---------------- Zm00001d004308\n",
       " ...   ... ...\n",
       "[51]   772 -----------------------...DGKA------------------ HORVU5Hr1G106160\n",
       "[52]   772 -----------------------...ELPVPDSKARP----------- HORVU7Hr1G031650\n",
       "[53]   772 -----------------------...---------------------- HORVU7Hr1G098820\n",
       "[54]   772 -----------------------...---------------------- Zm00001d023939\n",
       "[55]   772 -----------------------...---------------------- HORVU5Hr1G028030"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#readAAStringSet(paste0(\"tea/\", 1, \"/muscle/\", \"N0.HOG0000207\", \".afa\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hypotheses <- read_delim(\"hypotheses.tsv\", delim = \"\\t\")\n",
    "\n",
    "\n",
    "#create hypotheses object\n",
    "# each object has list of exp. OGs\n",
    "#iterate over list of hyptheses and associate expansions, fasta, msa and trees\n",
    "\n",
    "\n",
    "#first goal - hypothesis object with hypothesis and all exp.HOGs\n",
    "\n",
    "#could do it over objects in tea directory\n",
    "#list.files(path = \"tea\", pattern = \"*\", full.names=FALSE)\n",
    "\n",
    "\n",
    "#alternatively via parsing of hyptheses.tsv\n",
    "hypotheses$hypothesis\n",
    "\n",
    "\n",
    "## define three classes\n",
    "# class for the expanded_OG - containing all different types of data we have on it\n",
    "setClass(\"expanded_OG\", slots=list(genes=\"spec_tbl_df\",\n",
    "                                   blast_table=\"tbl_df\",\n",
    "                                   num_genes_HOG=\"numeric\",\n",
    "                                   num_genes_extend=\"numeric\",\n",
    "                                   num_genes_complete=\"numeric\",\n",
    "                                   genes_HOG=\"tbl_df\",\n",
    "                                   genes_extend_hits=\"tbl_df\",\n",
    "                                   fasta_files=\"list\", \n",
    "                                   msa=\"AAStringSet\", \n",
    "                                   tree=\"phylo\"))\n",
    "\n",
    "\n",
    "# class for the hypotheses\n",
    "# adding a prototype is essential here for ortho_intersect_plot since gg has to be defined as \n",
    "# register an old-style S3 class using setOldClass\n",
    "#https://stackoverflow.com/questions/12636056/why-sometimes-i-cant-set-a-class-definition-as-slot-in-a-s4-class\n",
    "#https://stackoverflow.com/questions/13841400/use-s3-virtual-class-as-slot-of-an-s4-class-got-error-got-class-s4-should-b\n",
    "#had trouble getting this to work; thus solved differently\n",
    "# -> seperate list object \"Ortho_intersect_plots\" hypothesis specific access via index\n",
    "setClass(\"hypothesis\", \n",
    "#         prototype=prototype(ortho_intersect_plot=structure(list(), class=\"gList\")),\n",
    "         slots=list(description=\"character\", \n",
    "                                  number=\"character\",\n",
    "                                  expanded_in =\"character\", \n",
    "                                  compared_to=\"character\", \n",
    "                                  expanded_OGs=\"list\",\n",
    "                                  species_tree=\"phylo\"))\n",
    "#                                  ortho_intersect_plot=\"gg\"))\n",
    "\n",
    "# class for extended BLAST hits info\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\",\n",
    "                    genes_extend_hits=\"tbl_df\")\n",
    "         )\n",
    "\n",
    "\n",
    "rm(t)\n",
    "rm(hypothesis_1)\n",
    "rm(hypothesis_2)\n",
    "rm(HYPOTHESES.a2tea)\n",
    "\n",
    "\n",
    "#remove protein_names in the snakemake pipeline - directories clean enough\n",
    "\n",
    "\n",
    "for (hypothesis in hypotheses$hypothesis) {\n",
    "    # read-in extended_BLAST_hits.RDS object of hypothesis\n",
    "    extended_BLAST_hits <- readRDS(paste0(\"tea/\", hypothesis, \"/extended_BLAST_hits/extended_BLAST_hits.RDS\"))\n",
    "    \n",
    "    # print(paste0(\"hypothesis_\", hypo))\n",
    "    \n",
    "    # create empty list object for hypothesis #\n",
    "    assign(paste0(\"hypothesis_\", hypothesis), list())\n",
    "\n",
    "    # assign list of names\n",
    "    expanded_OGs <- list.files(path = paste0(\"tea/\", hypothesis, \"/exp_OGs_proteinnames/\"), \n",
    "                               pattern = \"*\", \n",
    "                               full.names=FALSE)\n",
    "    expanded_OGs_short <- str_sub(expanded_OGs, end=-5)\n",
    "\n",
    "    # assign or subset the orthofinder species_tree based on the current hypothesis\n",
    "    speciesTree <- ape::keep.tip(read.tree(\"orthofinder/final-results/Species_Tree/SpeciesTree_rooted_node_labels.txt\"), \n",
    "                                 c(unlist(str_split(hypotheses$expanded_in[hypothesis], \";\")), \n",
    "                                   unlist(str_split(hypotheses$compared_to[hypothesis], \";\"))))\n",
    "    \n",
    "    \n",
    "    #create empty list\n",
    "    t <- list()\n",
    "\n",
    "    for (exp_OG in expanded_OGs_short) {\n",
    "    # print(exp_OG)\n",
    "        test <- new(\"expanded_OG\", \n",
    "             genes=read_table(paste0(\"tea/\", hypothesis, \"/exp_OGs_proteinnames/\", exp_OG, \".txt\"), col_names = FALSE),\n",
    "             blast_table=extended_BLAST_hits[[exp_OG]]@blast_table,\n",
    "             num_genes_HOG=extended_BLAST_hits[[exp_OG]]@num_genes_HOG,\n",
    "             num_genes_extend=extended_BLAST_hits[[exp_OG]]@num_genes_extend,\n",
    "             num_genes_complete=extended_BLAST_hits[[exp_OG]]@num_genes_complete,\n",
    "             genes_HOG=extended_BLAST_hits[[exp_OG]]@genes_HOG,\n",
    "             genes_extend_hits=extended_BLAST_hits[[exp_OG]]@genes_extend_hits,\n",
    "             fasta_files=read.fasta(paste0(\"tea/\", hypothesis, \"/fa_records/\", exp_OG,\".fa\"), seqtype = \"AA\", as.string = TRUE), \n",
    "             msa=readAAStringSet(paste0(\"tea/\", hypothesis, \"/muscle/\", exp_OG, \".afa\")), \n",
    "             tree=read.tree(paste0(\"tea/\", hypothesis, \"/trees/\", exp_OG, \".tree\")))\n",
    "   \n",
    "        x <- list(test)\n",
    "        names(x) <- paste0(exp_OG)\n",
    "        t <- c(t, x)\n",
    "    }\n",
    " \n",
    "    # create for each hypothesis a complete hypothesis object with correct naming\n",
    "    # order: hypothesis@info (name, number, expand, compared, exp OGs)$indiv. exp_OGs@objects (exp.OG list, fa, msa, tree)\n",
    "    # \"expanded_in\" and \"compared_to\" are split on \";\" in case their size is > 1\n",
    "    assign(paste0(\"hypothesis_\", hypothesis),\n",
    "              new(\"hypothesis\", \n",
    "                  description=hypotheses$name[hypothesis],\n",
    "                  number=as.character(hypothesis),\n",
    "                  expanded_in=unlist(str_split(hypotheses$expanded_in[hypothesis], \";\")),\n",
    "                  compared_to=unlist(str_split(hypotheses$compared_to[hypothesis], \";\")), \n",
    "                  expanded_OGs=t,\n",
    "                  species_tree=speciesTree))#,\n",
    "#                  ortho_intersect_plot=NULL))\n",
    "}\n",
    "\n",
    "\n",
    "# create empty list for final complete hypothesis object\n",
    "\n",
    "#rm\n",
    "\n",
    "# create name list of all hypotheses\n",
    "hypotheses_list <- ls(pattern = \"hypothesis_\")\n",
    "\n",
    "# create empty list object to completely hold all hypotheses and associated data\n",
    "HYPOTHESES.a2tea <- list()\n",
    "\n",
    "for (hypothesis in hypotheses_list) {\n",
    "    h <- list(get(hypothesis))\n",
    "    names(h) <- paste0(hypothesis)\n",
    "    HYPOTHESES.a2tea <- c(HYPOTHESES.a2tea, h)\n",
    "}\n",
    "\n",
    "rm(hypothesis_1)\n",
    "rm(hypothesis_2)\n",
    "\n",
    "\n",
    "# final object is called:\n",
    "# HYPOTHESES.a2tea\n",
    "\n",
    "\n",
    "#add expanded OGs as column\n",
    "#    add_column(get(paste0(\"hypothesis_\", hypo)), expanded_OGs)\n",
    "#class(hypothesis_1@expanded_OGs) <- c(\"expanded_OG\", class(hypothesis_1@expanded_OGs))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "length(hypotheses$hypothesis)\n",
    "length(HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000027@genes$X1)\n",
    "length(HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000027@fasta_files)\n",
    "HYPOTHESES.a2tea$hypothesis_2@expanded_OGs$N0.HOG0000028@blast_table\n",
    "#?setClass"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### HOG level tables -needs to be behind creation of hypotheses, because they are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create hypothesis specfic HOG level file\n",
    "\n",
    "#find all tibbles -list\n",
    "#list.files(path = \"tea/1/expansion_tibble\", pattern = \"expansion_tibble.rds\", full.names=TRUE)\n",
    "\n",
    "\n",
    "#length(hypotheses_list)\n",
    "\n",
    "\n",
    "HOG_level_list <- list()\n",
    "\n",
    "#for (i in 1:length(hypotheses_list)) {\n",
    "for (i in 1:length(hypotheses$hypothesis)) {\n",
    "    assign(paste0(\"hypothesis_\", i, \"_expansion_tibble\"),\n",
    "           readRDS(\n",
    "               list.files(path = paste0(\"tea/\", i, \"/expansion_tibble\"), \n",
    "                      pattern = \"expansion_tibble.rds\", \n",
    "                      full.names=TRUE)\n",
    "                  )\n",
    "           )\n",
    "    # add to list and name list entries according to hypothesis\n",
    "    hog_level <- list(get(paste0(\"hypothesis_\", i, \"_expansion_tibble\")))\n",
    "    names(hog_level) <- paste0(\"hypothesis_\", i)\n",
    "    HOG_level_list <- c(HOG_level_list, hog_level)\n",
    "}\n",
    "\n",
    "#e.g. the following displays the tibble\n",
    "#hypothesis_2_expansion_tibble\n",
    "\n",
    "#r\n",
    "#append \"_gene_count\" to all species in HOG tables\n",
    "#get number of significantly regulated genes per HOG (per species and total) \n",
    "# -> (hmm perhaps do this once and then merge?)\n",
    "\n",
    "\n",
    "# get count of species, HOG, significant combination from HOG_DE.a2tea\n",
    "# reduce it to HOG, species, count\n",
    "sig_genes_per_species_and_HOG <- HOG_DE.a2tea %>% \n",
    "                                     filter(significant == c(\"yes\")) %>%\n",
    "                                     filter(HOG != c(\"singleton\")) %>%\n",
    "                                        group_by(HOG, species, significant) %>% \n",
    "                                        mutate(count = n()) %>%\n",
    "                                        ungroup() %>%\n",
    "                                        select(HOG, species, count) %>%\n",
    "            # https://www.r-bloggers.com/2018/05/workaround-for-tidyrspread-with-duplicate-row-identifiers/\n",
    "            # spread error when no indexing for data\n",
    "                                            group_by(species) %>% \n",
    "                                            mutate(grouped_id = row_number()) %>%  \n",
    "                                            spread(species, count) %>% \n",
    "                                            select(-grouped_id) %>%\n",
    "            # easy workaround for duplicated rows\n",
    "                                                distinct()%>% \n",
    "            # rename species columns containing now the counts of sig. DE genes\n",
    "                                                    rename_at(vars(-HOG), ~ paste0(., '_sigDE'))\n",
    "    # and also add a column summing the rowwise sig. DE counts for all species\n",
    "    total_sigDE <- sig_genes_per_species_and_HOG %>% select(-HOG) %>% rowSums(na.rm = TRUE) \n",
    "    sig_genes_per_species_and_HOG <- add_column(sig_genes_per_species_and_HOG, total_sigDE)\n",
    "    sig_genes_per_species_and_HOG <- sig_genes_per_species_and_HOG  %>%\n",
    "                                         group_by(HOG) %>%\n",
    "                                         # mutate all NAs to 0s\n",
    "                                         mutate_at(vars(-group_cols()), ~replace(., is.na(.), 0)) %>%\n",
    "                                         # merge rows per HOG - results in one line per HOG\n",
    "                                         #summarise_all(funs(sum))\n",
    "                                         summarise_all(list(sum))\n",
    "\n",
    "# change the zeros back to NAs\n",
    "    sig_genes_per_species_and_HOG <- sig_genes_per_species_and_HOG  %>%\n",
    "                                         na_if(0)\n",
    "\n",
    "#sig_genes_per_species_and_HOG\n",
    "#total_sigDE\n",
    "#inner_join(HOG_level_list[[1]], sig_genes_per_species_and_HOG, by = c(\"HOG\"))\n",
    "#full_join important - better to keep all HOG rows and allow filtering dynamically in the WebAPP itself\n",
    "\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "    HOG_level_list[[i]] <- HOG_level_list[[i]] %>% rename_at(vars(-HOG, -expansion), ~ paste0(., '_total'))\n",
    "    HOG_level_list[[i]] <- full_join(HOG_level_list[[i]], sig_genes_per_species_and_HOG, by = c(\"HOG\"))\n",
    "}\n",
    "\n",
    "str(HOG_level_list)\n",
    "\n",
    "\n",
    "#head(HOG_level_list)\n",
    "\n",
    "#df %>% rename_at(vars(-Name, -State), ~ paste0(., '_2017'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(HYPOTHESES.a2tea)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HOG_level_list$hypothesis_1\n",
    "HOG_level_list$hypothesis_1 %>%\n",
    "    filter(HOG == \"N0.HOG0000318\")\n",
    "\n",
    "#HOG_level_list$hypothesis_1 %>% group_by(HOG) %>% summarise(sum(total_sigDE))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### creating Venn (or alternative diagrammes for the hypothesis)\n",
    "### -> will be moved to the WebApp and computed there..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create this awesome function to change area of plotting\n",
    "# https://stackoverflow.com/questions/1279003/specify-width-and-height-of-plot\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "# use funtion to set area a bit wider\n",
    "set_plot_dimensions(32, 10)\n",
    "\n",
    "\n",
    "\n",
    "#step 1: removing all species not in hypothesis or keep only those that are\n",
    "# using HOG_level_list and looking at the \"_total\" columns\n",
    "#  only species in this hypothesis are in here ;D\n",
    "# -> still drop all columns not necessary?\n",
    "#step 2: perform check on size of this set\n",
    "# -> if < 5 then create Venn diagramme\n",
    "# -> if > 5 then create alternative plot\n",
    "# create plot and safe as sub element in HYPOTHESES.a2tea object \n",
    "\n",
    "\n",
    "\n",
    "#HOG_DE.a2tea\n",
    "#HOG_level_list[[1]]\n",
    "#choose based on set length 2,3,4,5 after that UpsetR\n",
    "\n",
    "\n",
    "Ortho_intersect_plots <- list()\n",
    "\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "\n",
    "workset <-  HOG_level_list[[i]]\n",
    "# drop columns that are unnecessary\n",
    "workset <- workset %>% select(HOG, ends_with(\"_total\")) \n",
    "# add this point we can easily compute the number of species/ecotypes etc. \n",
    "# #species = (columns of current workset - 1) \n",
    "species_num <- ncol(workset) - 1\n",
    "# change 0 to NAs since we can more easily filter on that\n",
    "is.na(workset) <- workset==0\n",
    "# remove '_total' from column names\n",
    "#str_remove(colnames(workset), \"_total\")\n",
    "workset <- workset %>% setNames(names(workset) %>% stringr::str_replace(\"_total\",\"\"))\n",
    "\n",
    "# for testing purposes - remove Genotype vectors\n",
    "rm(Genotype1,Genotype2,Genotype3)\n",
    "\n",
    "genotypeList <- list()\n",
    "for (j in 1:species_num) {\n",
    "    assign(paste0(\"Genotype\", j), as.vector(simplify_all(workset[!is.na(workset[j+1]),\"HOG\"])[[1]]))\n",
    "    subList <- list(get(paste0(\"Genotype\", j)))\n",
    "    names(subList) <- colnames(workset)[j+1]\n",
    "    genotypeList <- c(genotypeList, subList)\n",
    "}\n",
    "\n",
    "\n",
    "#using UpSetR\n",
    "upsetOrtho <- upset(fromList(genotypeList), \n",
    "                   order.by = \"freq\",\n",
    "                   empty.intersections = \"on\",\n",
    "                   point.size = 3.5, \n",
    "                   line.size = 2, \n",
    "                   mainbar.y.label = \"HOG Intersections\", \n",
    "                   text.scale = c(2, 1.3, 1, 1, 2, 2))\n",
    "\n",
    "p <- upsetOrtho\n",
    "\n",
    "# if we have less than 5 species draw a venn diagramme\n",
    "# combine both plots with some trickery\n",
    "if (species_num < 5) {\n",
    "    \n",
    "# draw venn diagramme\n",
    "orthoVenn <- venn.diagram(genotypeList,\n",
    "                           print.mode = c(\"raw\",\"percent\"),\n",
    "                           sigdigs = 1,\n",
    "                           filename = NULL,\n",
    "                           category = names(genotypeList),\n",
    "                           fill = rainbow(length(names(genotypeList))),\n",
    "                           cat.col = rainbow(length(names(genotypeList))),\n",
    "                           cat.cex = 2,\n",
    "                           cex = 1.5,\n",
    "                           margin = 0.05\n",
    "                           )    \n",
    "    \n",
    "# as.grob from ggplotify is necessary\n",
    "orthoIntersect <- plot_grid(orthoVenn, \n",
    "          as.grob(upsetOrtho), \n",
    "          rel_widths = c(1/3, 2/3))\n",
    "\n",
    "#p <- ggplotify::as.ggplot(orthoIntersect)\n",
    "p <- orthoIntersect\n",
    "    }\n",
    "\n",
    "Ortho_intersect_plots[[i]] <- p\n",
    "\n",
    "}\n",
    "    \n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Ortho_intersect_plots[[1]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### computation of tea value (first idea)\n",
    "\n",
    "expansion ratio (exp/com) (9/4) x expression ratio (com/exp) (2/6) \n",
    "---- don't need (n_species_expanded/n_species_compared); stays equal in a hypothesis; similar to p-values not          comparable between different experiments\n",
    "divided by HOG size (all genes in HOG for this particular hypothesis) x number of significantly regulated genes\n",
    "\n",
    "\n",
    "problem - how to deal with missing values???"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# should I only have hypothesis species (total count and sigDE) HOG_level_list tables?\n",
    "\n",
    "#new column tea-value\n",
    "#if expansion yes continue; else NA value\n",
    "#because we set expansion as criterium, species gene count has to be >=1\n",
    "#sig DE counts can be NA ! - we could just add a pseudocount of 1 at these positions for the calculation\n",
    "\n",
    "#so per line that has expansion = yes:\n",
    "#get all expanded species; sum their gene counts; sum their sigDE counts\n",
    "#get all compared species; sum their gene counts; sum their sigDE counts\n",
    "#expansion_ratio <- divide summed gene count expanded by summed gene count compared\n",
    "#expression_ratio <- divide summed SigDE count expanded by summed SigDE count compared\n",
    "#e2_ratio <- divide expansion_ratio by expression_ratio\n",
    "#tea-value <- divide e2_ratio by total number of genes in HOG and then by total number of SigDE in HOG\n",
    "\n",
    "\n",
    "# outer most loop; going through hypotheses in HOG_level_list\n",
    "for (i in 1:length(HOG_level_list)) {\n",
    "    \n",
    "    # access the names of all species important for the hypothesis - expanded_in; compared_to\n",
    "    print(c(unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\")), \n",
    "            unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\"))))\n",
    "    \n",
    "    # creating a relevant species vector might make this more future proof;\n",
    "    # since I might change the completeness of the HOG_level_list elements\n",
    "#    relevant_species <- c(unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\")), \n",
    "#                          unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\")))\n",
    "    \n",
    "    expanded_species <- unlist(str_split(HYPOTHESES.a2tea[[i]]@expanded_in, \";\"))\n",
    "    \n",
    "    compared_species <- unlist(str_split(HYPOTHESES.a2tea[[i]]@compared_to, \";\"))                              \n",
    "                                  \n",
    "    ## pre-allocate empty vector for tea value with length of nrow of current hypothesis of HOG_level_list\n",
    "    # neat; because NAs already in there so I can just skip if the HOG shows no expansion ;D\n",
    "    tea_value <- rep(NA, nrow(HOG_level_list[[i]]))\n",
    "    \n",
    "    \n",
    "    ## create \"geneCount_workset\" containing HOGs and counts of all relevant species\n",
    "    geneCount_workset <-  HOG_level_list[[i]]\n",
    "\n",
    "    # create geneCount_EXP_workset & geneCount_COMP_workset + drop columns that are unnecessary                  \n",
    "    geneCount_EXP_workset <- geneCount_workset %>% \n",
    "                             select(ends_with(\"_total\")) %>%\n",
    "                             select(contains(expanded_species))\n",
    "                        # also remove the \"_total\" ending\n",
    "    geneCount_EXP_workset <- geneCount_EXP_workset %>% \n",
    "                             setNames(names(geneCount_EXP_workset) %>% \n",
    "                             stringr::str_replace(\"_total\",\"\"))\n",
    "                          \n",
    "    geneCount_COMP_workset <- geneCount_workset %>% \n",
    "                             select(ends_with(\"_total\")) %>%\n",
    "                             select(contains(compared_species))\n",
    "                        # also remove the \"_total\" ending\n",
    "    geneCount_COMP_workset <- geneCount_COMP_workset %>% \n",
    "                             setNames(names(geneCount_COMP_workset) %>% \n",
    "                             stringr::str_replace(\"_total\",\"\"))\n",
    "                    \n",
    "\n",
    "    ## create \"geneSig_workset\" containing HOGs and counts of all relevant species \n",
    "    geneSig_workset <-  HOG_level_list[[i]]\n",
    "    \n",
    "    # create geneSig_EXP_workset & geneSig_COMP_workset + drop columns that are unnecessary                  \n",
    "    geneSig_EXP_workset <- geneSig_workset %>% \n",
    "                             select(ends_with(\"_sigDE\")) %>%\n",
    "                             select(contains(expanded_species))\n",
    "                        # also remove the \"_sigDE\" ending\n",
    "    geneSig_EXP_workset <- geneSig_EXP_workset %>% \n",
    "                             setNames(names(geneSig_EXP_workset) %>% \n",
    "                             stringr::str_replace(\"_sigDE\",\"\"))\n",
    "                          \n",
    "    geneSig_COMP_workset <- geneSig_workset %>% \n",
    "                             select(ends_with(\"_sigDE\")) %>%\n",
    "                             select(contains(compared_species))\n",
    "                        # also remove the \"_sigDE\" ending\n",
    "    geneSig_COMP_workset <- geneSig_COMP_workset %>% \n",
    "                             setNames(names(geneSig_COMP_workset) %>% \n",
    "                             stringr::str_replace(\"_sigDE\",\"\"))\n",
    "\n",
    "                               \n",
    "    # nested loop that goes through current hypothesis HOG_level_list line by line\n",
    "    for (j in 1:nrow(HOG_level_list[[i]])) {\n",
    "        # write value into index slot if current HOG does pass hard filter expansion criterium\n",
    "        # if expansion == \"no\" do nothing; since we prefilled all positions with NA\n",
    "        if (HOG_level_list[[i]][j,\"expansion\"] == \"yes\") {\n",
    "            \n",
    "            ## compute all the different sub-values for the tea_value\n",
    "            # count of all genes in expanded species in HOG\n",
    "            geneCount_EXP_HOG <- geneCount_EXP_workset[j,] %>%\n",
    "                                 dplyr::rowwise() %>%\n",
    "                                 sum()\n",
    "            \n",
    "            # count of all genes in compared species in HOG\n",
    "            geneCount_COMP_HOG <- geneCount_COMP_workset[j,] %>%\n",
    "                                      dplyr::rowwise() %>%\n",
    "                                      sum()\n",
    "            \n",
    "            # sum genes of expanded & compared species in HOG\n",
    "            geneCount_BOTH_HOG <- sum(geneCount_EXP_HOG, geneCount_COMP_HOG)\n",
    "            \n",
    "            \n",
    "            # count of all sig DE genes in expanded species in HOG\n",
    "            geneSig_EXP_HOG <- geneSig_EXP_workset[j,] %>%\n",
    "                                 dplyr::rowwise() %>%\n",
    "                                 sum()\n",
    "            \n",
    "            # count of all sig DE genes in compared species in HOG\n",
    "            geneSig_COMP_HOG <- geneSig_COMP_workset[j,] %>%\n",
    "                                      dplyr::rowwise() %>%\n",
    "                                      sum()\n",
    "            \n",
    "            ## sigDE columns can contain zero (NA in table) counts\n",
    "            ## we exchange this for a pseudocount 1\n",
    "            ## doesn't impact tea_value too much\n",
    "            if (is.na(geneSig_EXP_HOG)) {\n",
    "                geneSig_EXP_HOG <- 1\n",
    "            }\n",
    "            \n",
    "            if (is.na(geneSig_COMP_HOG)) {\n",
    "                geneSig_COMP_HOG <- 1\n",
    "            }\n",
    "            \n",
    "            # sum Sig DE genes of expanded & compared species in HOG\n",
    "            geneSig_BOTH_HOG <- sum(geneSig_EXP_HOG, geneSig_COMP_HOG)\n",
    "\n",
    "            \n",
    "            ## perform tea value calculation\n",
    "            int_tea_value <- (geneCount_EXP_HOG / geneCount_COMP_HOG) * (geneSig_COMP_HOG / geneSig_EXP_HOG) / \n",
    "                                                  (geneCount_BOTH_HOG * geneSig_BOTH_HOG)\n",
    "            \n",
    "            # lastly, add computed value to tea_value at index of current j loop position\n",
    "            tea_value[j] <- int_tea_value\n",
    "        }\n",
    "    }\n",
    "    \n",
    "    ## add tea_value as new column to HOG_level_list table\n",
    "    HOG_level_list[[i]] <- HOG_level_list[[i]] %>% \n",
    "                               add_column(tea_value, .after = \"HOG\")\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "HOG_level_list$hypothesis_2 %>%\n",
    "    arrange(HOG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#c(unlist(str_split(HYPOTHESES.a2tea[[2]]@expanded_in, \";\")), \n",
    "#  unlist(str_split(HYPOTHESES.a2tea[[2]]@compared_to, \";\")))\n",
    "#geneSig_workset <-  HOG_level_list[[1]]\n",
    "#names(geneSig_workset)\n",
    "#head(geneSig_workset %>% setNames(names(geneSig_workset) %>% stringr::str_replace(\"_sigDE\",\"\")))\n",
    "#HOG_level_list[[i]][1,\"expansion\"]\n",
    "#tea_value[2] <- \"asdasdd\"\n",
    "str(tea_value)\n",
    "head(geneCount_EXP_workset[2,])\n",
    "\n",
    "HOG_level_list[[1]][3,c(2,3)] %>%\n",
    "    dplyr::rowwise() %>%\n",
    "    sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(HYPOTHESES.a2tea)\n",
    "#HOG_level_list[[2]] %>% add_column(tea_value, .after = \"HOG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HYPOTHESES.a2tea$hypothesis_2@species_tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### adding the computed cafe p-values to the HOG_level tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for (hypothesis_num in 1:length(HOG_level_list)) {\n",
    "\n",
    "  cafe_tibble <- read_tsv(paste0(\"cafe/\", hypothesis_num, \"/cafe_complete_results/Gamma_family_results.txt\"))[1:2]\n",
    "\n",
    "  HOG_level_list[[hypothesis_num]] <- HOG_level_list[[hypothesis_num]] %>%\n",
    "                                        left_join(., \n",
    "                                                  cafe_tibble, \n",
    "                                                  by = c(\"HOG\" = \"#FamilyID\")\n",
    "                                                 ) %>%\n",
    "                                        rename(cafe_pvalue = pvalue) %>%\n",
    "                                        relocate(cafe_pvalue, .after = tea_value) %>%\n",
    "                                        arrange(tea_value)\n",
    "}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "head(HOG_level_list$hypothesis_1)\n",
    "head(HOG_level_list$hypothesis_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Last step: saving everything to one file which is input for the A2TEA WebApp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(hypotheses, \n",
    "     HYPOTHESES.a2tea, \n",
    "     HOG_DE.a2tea, \n",
    "     HOG_level_list, \n",
    "#     Ortho_intersect_plots,\n",
    "     file = \"tea/A2TEA_finished.RData\")\n",
    "\n",
    "# To load the data again\n",
    "#load(\"tea/A2TEA_finished.RData\")\n",
    "\n",
    "#perhaps use this to name output based on time?\n",
    "#paste0(\"tea/A2TEA_finished_\",\n",
    "#    str_replace_all(\n",
    "#    str_replace_all(\n",
    "#        Sys.time(), \":\", \"-\"),\n",
    "#    \" \", \"_\"\n",
    "#                    ), \".RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "load(\"tea/A2TEA_finished.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "HOG_level_list$hypothesis_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "HYPOTHESES.a2tea$hypothesis_1@expanded_OGs$N0.HOG0000207@blast_table"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
