{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "R.version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(\"ggtree\")\n",
    "\n",
    "#install.packages(\"tidyverse\")\n",
    "#install.packages(\"readr\")\n",
    "\n",
    "\n",
    "#install.packages(\"dplyr\")\n",
    "\n",
    "\n",
    "#not\n",
    "####install.packages(\"stringr\")\n",
    "#remove.packages(\"ggtree\")\n",
    "#remove.packages(\"readr\")\n",
    "#remove.packages(\"dplyr\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:plyr’:\n",
      "\n",
      "    arrange, count, desc, failwith, id, mutate, rename, summarise,\n",
      "    summarize\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "###library(ggtree)\n",
    "#library(readr)\n",
    "#plyr before dplyr!\n",
    "#library(plyr)\n",
    "#library(dplyr)\n",
    "#library(tidyverse)\n",
    "#library(stringr)\n",
    "library(readr)\n",
    "library(plyr)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "library(tidyr)\n",
    "library(tibble)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read-in \"name\", \"expanded_in\" and \"compared_to\" \n",
    "num = \"2\"\n",
    "name = \"Expanded in barley compared to maize\"\n",
    "#expanded_in = \"Arabidopsis_thaliana\"\n",
    "#compared_to = c(\"Hordeum_vulgare\", \"Zea_mays\")\n",
    "#expanded_in = \"Arabidopsis_thaliana\"\n",
    "#compared_to = c(\"Zea_mays\", \"Hordeum_vulgare\")\n",
    "expanded_in = c(\"Zea_mays\", \"Hordeum_vulgare\")\n",
    "compared_to = \"Arabidopsis_thaliana\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[36m──\u001b[39m \u001b[1m\u001b[1mColumn specification\u001b[1m\u001b[22m \u001b[36m───────────────────────────────────────────────\u001b[39m\n",
      "cols(\n",
      "  Orthogroup = \u001b[31mcol_character()\u001b[39m,\n",
      "  Arabidopsis_thaliana = \u001b[32mcol_double()\u001b[39m,\n",
      "  Hordeum_vulgare = \u001b[32mcol_double()\u001b[39m,\n",
      "  Zea_mays = \u001b[32mcol_double()\u001b[39m,\n",
      "  Total = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "OG.GC <- readr::read_tsv(\"../orthofinder/final-results/Orthogroups/Orthogroups.GeneCount.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'Arabidopsis_thaliana'"
      ],
      "text/latex": [
       "'Arabidopsis\\_thaliana'"
      ],
      "text/markdown": [
       "'Arabidopsis_thaliana'"
      ],
      "text/plain": [
       "[1] \"Arabidopsis_thaliana\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#THis step is actually in the python functions; however here I I need a solution to test the downstream steps:\n",
    "#I can use strsplit to split the \";\" seperated columns\n",
    "#strsplit creates a list and the [[1]] selects the first list item (we only have one, in this case).\n",
    "#The result at this point is in list form, we can use \"unlist\" that works with the downstream steps.\n",
    "#compared_to\n",
    "#c_t_species <- unlist(strsplit(compared_to, \";\"))\n",
    "#compared_to\n",
    "c_t_species <- compared_to\n",
    "c_t_species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error: Problem with `filter()` input `..1`.\n\u001b[34mℹ\u001b[39m Input `..1` is `get(expanded_in) > 3 * (get(i))`.\n\u001b[31m✖\u001b[39m first argument has length > 1\n",
     "output_type": "error",
     "traceback": [
      "Error: Problem with `filter()` input `..1`.\n\u001b[34mℹ\u001b[39m Input `..1` is `get(expanded_in) > 3 * (get(i))`.\n\u001b[31m✖\u001b[39m first argument has length > 1\nTraceback:\n",
      "1. expanded_OGs %>% dplyr::filter(get(expanded_in) > 3 * (get(i)))",
      "2. dplyr::filter(., get(expanded_in) > 3 * (get(i)))",
      "3. filter.data.frame(., get(expanded_in) > 3 * (get(i)))",
      "4. filter_rows(.data, ..., caller_env = caller_env())",
      "5. withCallingHandlers({\n .     dots <- new_quosures(flatten(imap(dots, function(dot, index) {\n .         env_filter$current_expression <- index\n .         expand_if_across(dot)\n .     })))\n .     mask$eval_all_filter(dots, env_filter)\n . }, error = function(e) {\n .     local_call_step(dots = dots, .index = env_filter$current_expression, \n .         .fn = \"filter\")\n .     abort(c(cnd_bullet_header(), i = cnd_bullet_input_info(), \n .         x = conditionMessage(e), i = cnd_bullet_cur_group_label()), \n .         class = \"dplyr_error\")\n . })",
      "6. mask$eval_all_filter(dots, env_filter)",
      "7. get(expanded_in)",
      "8. .handleSimpleError(function (e) \n . {\n .     local_call_step(dots = dots, .index = env_filter$current_expression, \n .         .fn = \"filter\")\n .     abort(c(cnd_bullet_header(), i = cnd_bullet_input_info(), \n .         x = conditionMessage(e), i = cnd_bullet_cur_group_label()), \n .         class = \"dplyr_error\")\n . }, \"first argument has length > 1\", base::quote(get(expanded_in)))",
      "9. h(simpleError(msg, call))",
      "10. abort(c(cnd_bullet_header(), i = cnd_bullet_input_info(), x = conditionMessage(e), \n  .     i = cnd_bullet_cur_group_label()), class = \"dplyr_error\")",
      "11. signal_abort(cnd)"
     ]
    }
   ],
   "source": [
    "#get() function solves my problems of using the character functions inside dplyr\n",
    "#expanded_OGs <- OG.GC\n",
    "#for (i in c_t_species) {\n",
    "#    expanded_OGs <- expanded_OGs %>% dplyr::filter(get(expanded_in) > 3*(get(i)))\n",
    "#    }\n",
    "#expanded_OGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"hello\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##keeping in line with smaller scripts and modularity:\n",
    "\n",
    "#1) save Orthogroups for this hypothesis as first result; TEA directory; subd. GFEs ;subd. hypothesis number\n",
    "\n",
    "#2)parallel seperate rule\n",
    "#getting all BLAST hits and substituting for actual OG names\n",
    "\n",
    "#3) use hypothesis wildcard and new R script to perform BLAST filtering\n",
    "# -> write these results to file again\n",
    "\n",
    "#4) snakemake muscle, gblocks and fastTree (alternative programmes?)\n",
    "\n",
    "#5) load data into R, plot with ggtree; \n",
    "#-> here, stability and cleanup\n",
    "\n",
    "#6) and then all the other stuff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#concatenate all BLAST seaches to one file and create parseable tibble with read_delim\n",
    "#could add that only BLAST searches of our species in this hypothesis are used - toDO for later\n",
    "#easiest and efficient way is to bind outside of the loop\n",
    "#do.call is a nice way rbind(bind_rows[[1]], bind_rows[[2]], ...)\n",
    "datalist = list()\n",
    "for (i in Sys.glob(\"../orthofinder/search/*.txt\")) {\n",
    "    datalist[[i]] <- readr::read_delim(file = i,\n",
    "                                delim = \"\\t\",\n",
    "                                col_names = c(\n",
    "                                    \"qseqid\",\n",
    "                                    \"sseqid\",\n",
    "                                    \"pident\",\n",
    "                                    \"length\",\n",
    "                                    \"mismatch\",\n",
    "                                    \"gapopen\",\n",
    "                                    \"qstart\",\n",
    "                                    \"qend\",\n",
    "                                    \"sstart\",\n",
    "                                    \"send\",\n",
    "                                    \"evalue\",\n",
    "                                    \"bitscore\"),\n",
    "                                col_types = c(\n",
    "                                    qseqid = col_character(),\n",
    "                                    sseqid = col_character(),\n",
    "                                    pident = col_double(),\n",
    "                                    length = col_double(),\n",
    "                                    mismatch = col_double(),\n",
    "                                    gapopen = col_double(),\n",
    "                                    qstart = col_double(),\n",
    "                                    qend = col_double(),\n",
    "                                    sstart = col_double(),\n",
    "                                    send = col_double(),\n",
    "                                    evalue = col_double(),\n",
    "                                    bitscore = col_double()\n",
    "                                                )\n",
    "                                )\n",
    "                                                }\n",
    "\n",
    "all_BLAST <- base::do.call(bind_rows, datalist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "head(all_BLAST)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read-in the conversion table: orthofinder sequence IDs and corresponding actual gene/protein names\n",
    "#with \"str_remove_all\" I also easily get rid of the \":\"\n",
    "conversion <- readr::read_delim(file = \"../orthofinder/SequenceIDs.txt\",\n",
    "                delim = \" \",\n",
    "                col_names = c(\"seqid\", \"name\"),\n",
    "                col_types = c(\n",
    "                   seqid = col_character(),\n",
    "                   name = col_character())               \n",
    "               )\n",
    "conversion <- conversion %>%  dplyr::mutate(seqid = str_remove_all(seqid, \":\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conversion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create new columns with the actual gene/protein names\n",
    "all_BLAST_reformatted <- all_BLAST %>% dplyr::left_join(conversion, by = c(\"qseqid\" = \"seqid\"))\n",
    "all_BLAST_reformatted <- all_BLAST_reformatted %>% dplyr::left_join(conversion, by = c(\"sseqid\" = \"seqid\"))\n",
    "#position after qseqid and sseqid respectively\n",
    "all_BLAST_reformatted <- all_BLAST_reformatted %>% dplyr::relocate(name.x, .after = qseqid)\n",
    "all_BLAST_reformatted <- all_BLAST_reformatted %>% dplyr::relocate(name.y, .after = sseqid)\n",
    "#rename to qseqid/sseqid_name\n",
    "all_BLAST_reformatted <- dplyr::rename(all_BLAST_reformatted,  qseqid_name = name.x)\n",
    "all_BLAST_reformatted <- dplyr::rename(all_BLAST_reformatted, sseqid_name = name.y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#write to file; for reference\n",
    "#write_delim(as.data.frame(foo), \"test.tsv\", delim = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read-in Orthogroups.txt old!!\n",
    "orthogroups <- readr::read_delim(file = \"../orthofinder/final-results/Orthogroups/Orthogroups.txt\",\n",
    "                          delim = \":\",\n",
    "                          col_names = c(\"OG\", \"genes\"),\n",
    "                          col_types = c(\n",
    "                              OG = col_character(),\n",
    "                              genes = col_character())\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[36m──\u001b[39m \u001b[1m\u001b[1mColumn specification\u001b[1m\u001b[22m \u001b[36m───────────────────────────────────────────────\u001b[39m\n",
      "cols(\n",
      "  HOG = \u001b[31mcol_character()\u001b[39m,\n",
      "  OG = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Gene Tree Parent Clade` = \u001b[31mcol_character()\u001b[39m,\n",
      "  Arabidopsis_thaliana = \u001b[31mcol_character()\u001b[39m,\n",
      "  Hordeum_vulgare = \u001b[31mcol_character()\u001b[39m,\n",
      "  Zea_mays = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>HOG</th><th scope=col>Zea_mays</th><th scope=col>Hordeum_vulgare</th><th scope=col>Arabidopsis_thaliana</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>N0.HOG0000000</td><td>421</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>N0.HOG0000001</td><td>  2</td><td>229</td><td>  0</td></tr>\n",
       "\t<tr><td>N0.HOG0000002</td><td>  0</td><td>  1</td><td>  1</td></tr>\n",
       "\t<tr><td>N0.HOG0000003</td><td>140</td><td>  1</td><td>  1</td></tr>\n",
       "\t<tr><td>N0.HOG0000004</td><td>  0</td><td>129</td><td>  0</td></tr>\n",
       "\t<tr><td>N0.HOG0000005</td><td>  0</td><td>  0</td><td>118</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " HOG & Zea\\_mays & Hordeum\\_vulgare & Arabidopsis\\_thaliana\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t N0.HOG0000000 & 421 &   0 &   0\\\\\n",
       "\t N0.HOG0000001 &   2 & 229 &   0\\\\\n",
       "\t N0.HOG0000002 &   0 &   1 &   1\\\\\n",
       "\t N0.HOG0000003 & 140 &   1 &   1\\\\\n",
       "\t N0.HOG0000004 &   0 & 129 &   0\\\\\n",
       "\t N0.HOG0000005 &   0 &   0 & 118\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 4\n",
       "\n",
       "| HOG &lt;chr&gt; | Zea_mays &lt;dbl&gt; | Hordeum_vulgare &lt;dbl&gt; | Arabidopsis_thaliana &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| N0.HOG0000000 | 421 |   0 |   0 |\n",
       "| N0.HOG0000001 |   2 | 229 |   0 |\n",
       "| N0.HOG0000002 |   0 |   1 |   1 |\n",
       "| N0.HOG0000003 | 140 |   1 |   1 |\n",
       "| N0.HOG0000004 |   0 | 129 |   0 |\n",
       "| N0.HOG0000005 |   0 |   0 | 118 |\n",
       "\n"
      ],
      "text/plain": [
       "  HOG           Zea_mays Hordeum_vulgare Arabidopsis_thaliana\n",
       "1 N0.HOG0000000 421        0               0                 \n",
       "2 N0.HOG0000001   2      229               0                 \n",
       "3 N0.HOG0000002   0        1               1                 \n",
       "4 N0.HOG0000003 140        1               1                 \n",
       "5 N0.HOG0000004   0      129               0                 \n",
       "6 N0.HOG0000005   0        0             118                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#read-in Orthogroups.txt new!!\n",
    "#expanded OG_s also have to be computed based on this..\n",
    "ph_orthogroups <- readr::read_delim(file = \"../orthofinder/final-results/Phylogenetic_Hierarchical_Orthogroups/N0.tsv\",\n",
    "                          delim = \"\\t\",\n",
    "#                          col_names = c(\"HOG\", \"genes\"),\n",
    "#                          col_types = c(\n",
    "#                              OG = col_character(),\n",
    "#                              genes = col_character())\n",
    "                         )\n",
    "#head(ph_orthogroups)\n",
    "#length(unlist(strsplit(ph_orthogroups[ph_orthogroups$HOG == \"N0.HOG0000103\",]$Arabidopsis_thaliana, \",\")))\n",
    "#length(unlist(strsplit(ph_orthogroups[\"N0.HOG0000103\",]$Arabidopsis_thaliana, \",\")))\n",
    "\n",
    "#get() function solves my problems of using the character functions inside dplyr\n",
    "\n",
    "#colnames(ph_orthogroups)\n",
    "#compared_to\n",
    "#length(unlist(strsplit(ph_orthogroups[ph_orthogroups[[compared_to]], \",\")))\n",
    "all_species <- c(expanded_in, compared_to)\n",
    "#all_species[1]\n",
    "#all_species[2]\n",
    "\n",
    "#create dataframe with numbers of genes per PHOG\n",
    "#we combine expanded_in and compared_to vectors to easily compute for everything we need\n",
    "\n",
    "HOG_df <- setNames(base::data.frame(matrix(ncol = 1 + length(all_species), nrow = 0)),\n",
    "                  c(\"HOG\", all_species))\n",
    "\n",
    "for (i in 1:nrow(ph_orthogroups)) {\n",
    "    \n",
    "    row = 0\n",
    "    \n",
    "    #print(ph_orthogroups[i,]$HOG)\n",
    "    row <- c(ph_orthogroups[i,]$HOG)\n",
    "    \n",
    "        for (j in c(all_species)) {\n",
    "            if (is.na(ph_orthogroups[i,][[j]]))\n",
    "            {\n",
    "                test = 0\n",
    "                row <- c(row, test)\n",
    "            }\n",
    "            else \n",
    "            {\n",
    "                test = length(unlist(strsplit(ph_orthogroups[i,][[j]], \",\")))\n",
    "                row <- c(row, test)\n",
    "            }\n",
    "}\n",
    "    HOG_df[i,] <- row\n",
    "} \n",
    "\n",
    "HOG_tibble <- as_tibble(HOG_df)\n",
    "\n",
    "for (k in 1:length(all_species)) {\n",
    "    o = all_species[k]\n",
    "    HOG_tibble[[o]] <- as.numeric(HOG_tibble[[o]])\n",
    "}\n",
    "head(HOG_tibble)\n",
    "\n",
    "#create \"copy\" as HOG_tibble is currently modified in the upcoming step\n",
    "HOG_tibble_complete <- HOG_tibble\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#head(my_tibble)\n",
    "#tail(my_tibble)\n",
    "#expanded_HOGs\n",
    "#head(ph_orthogroups[])\n",
    "HOG_tibble_complete[HOG_tibble_complete$HOG==\"N0.HOG0000783\",]\n",
    "HOG_tibble_complete[HOG_tibble_complete$HOG==\"N0.HOG0000027\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# function has to reduce input dataset; this way each iteration (compared species) is included\n",
    "# for all compared species first check for each HOG if expanded species is >= 3* \n",
    "# then keep rows in which expanded species has at least 3\n",
    "# since 3*1 = 3, we can simply choose to keep all rows with at least 3 in expanded_in since this \n",
    "## is performed after the other filtering ;D\n",
    "# this we should talk about... how to deal with 0 cases\n",
    "# in any case having this as user input shoud be the goal\n",
    "# with addition of the outer t loop now we can also have multiple expansion species ;D\n",
    "\n",
    "#for (i in c_t_species) {\n",
    "#        HOG_tibble <- HOG_tibble %>% dplyr::filter(get(expanded_in) >= 3*(get(i)))\n",
    "#        HOG_tibble <- HOG_tibble %>% dplyr::filter(get(expanded_in) >= 3)\n",
    "#        expanded_HOGs <- HOG_tibble\n",
    "#    }\n",
    "\n",
    "#for (t in expanded_in) {\n",
    "#for (i in c_t_species) {\n",
    "#        HOG_tibble <- HOG_tibble %>% dplyr::filter(get(expanded_in) >= expansion_factor*(get(i)))\n",
    "#        HOG_tibble <- HOG_tibble %>% dplyr::filter(get(expanded_in) >= expansion_factor)\n",
    "#    # this line will remove all HOGs with 0 cases in BOTH compared species\n",
    "#    # this means, that for now, a multispecies comparison demands that\n",
    "#    # for all compared_to species at least 1 gene has to be in the HOG!\n",
    "#        HOG_tibble <- HOG_tibble %>% dplyr::filter(get(i) > 0)\n",
    "#        expanded_HOGs <- HOG_tibble\n",
    "#    }\n",
    "#    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.1.0 (2021-05-18)\n",
       "Platform: x86_64-conda-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 20.04.2 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /data/home/stoecker/miniconda3/lib/libopenblasp-r0.3.15.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n",
       " [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n",
       " [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n",
       "[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] tibble_3.1.2  tidyr_1.1.3   stringr_1.4.0 dplyr_1.0.7   plyr_1.8.6   \n",
       "[6] readr_1.4.0  \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.7        pillar_1.6.1      compiler_4.1.0    base64enc_0.1-3  \n",
       " [5] tools_4.1.0       digest_0.6.27     uuid_0.1-4        jsonlite_1.7.2   \n",
       " [9] evaluate_0.14     lifecycle_1.0.0   pkgconfig_2.0.3   rlang_0.4.11     \n",
       "[13] IRdisplay_1.0     DBI_1.1.1         cli_2.5.0         rstudioapi_0.13  \n",
       "[17] IRkernel_1.2      repr_1.1.3        generics_0.1.0    vctrs_0.3.8      \n",
       "[21] hms_1.1.0         tidyselect_1.1.1  glue_1.4.2        R6_2.5.0         \n",
       "[25] fansi_0.5.0       pbdZMQ_0.3-5      purrr_0.3.4       magrittr_2.0.1   \n",
       "[29] ps_1.6.0          ellipsis_0.3.2    htmltools_0.5.1.1 assertthat_0.2.1 \n",
       "[33] utf8_1.2.1        stringi_1.7.3     crayon_1.4.1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo() "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "R version 4.1.0 (2021-05-18)\n",
    "Platform: x86_64-conda-linux-gnu (64-bit)\n",
    "Running under: Ubuntu 20.04.2 LTS\n",
    "\n",
    "Matrix products: default\n",
    "BLAS/LAPACK: /data/home/stoecker/miniconda3/lib/libopenblasp-r0.3.15.so\n",
    "\n",
    "locale:\n",
    " [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n",
    " [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n",
    " [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n",
    "[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n",
    "\n",
    "attached base packages:\n",
    "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
    "\n",
    "other attached packages:\n",
    "[1] tibble_3.1.2  tidyr_1.1.3   stringr_1.4.0 dplyr_1.0.7   plyr_1.8.6   \n",
    "[6] readr_1.4.0  \n",
    "\n",
    "loaded via a namespace (and not attached):\n",
    " [1] Rcpp_1.0.7        pillar_1.6.1      compiler_4.1.0    base64enc_0.1-3  \n",
    " [5] tools_4.1.0       digest_0.6.27     uuid_0.1-4        jsonlite_1.7.2   \n",
    " [9] evaluate_0.14     lifecycle_1.0.0   pkgconfig_2.0.3   rlang_0.4.11     \n",
    "[13] IRdisplay_1.0     DBI_1.1.1         cli_2.5.0         rstudioapi_0.13  \n",
    "[17] IRkernel_1.2      repr_1.1.3        generics_0.1.0    vctrs_0.3.8      \n",
    "[21] hms_1.1.0         tidyselect_1.1.1  glue_1.4.2        R6_2.5.0         \n",
    "[25] fansi_0.5.0       pbdZMQ_0.3-5      purrr_0.3.4       magrittr_2.0.1   \n",
    "[29] ps_1.6.0          ellipsis_0.3.2    htmltools_0.5.1.1 assertthat_0.2.1 \n",
    "[33] utf8_1.2.1        stringi_1.7.3     crayon_1.4.1    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 8 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>name</th><td>Expanded in Arabidopsis compared to Monocots</td><td>Expanded in barley compared to maize</td><td>bla                                 </td><td>bla2                    </td><td>bla3                                </td></tr>\n",
       "\t<tr><th scope=row>expanded_in</th><td>Arabidopsis_thaliana                        </td><td>Hordeum_vulgare                     </td><td>Zea_mays                            </td><td>Hordeum_vulgare;Zea_mays</td><td>Hordeum_vulgare;Zea_mays            </td></tr>\n",
       "\t<tr><th scope=row>compared_to</th><td>Zea_mays;Hordeum_vulgare                    </td><td>Zea_mays                            </td><td>Hordeum_vulgare;Arabidopsis_thaliana</td><td>Arabidopsis_thaliana    </td><td>Hordeum_vulgare;Arabidopsis_thaliana</td></tr>\n",
       "\t<tr><th scope=row>Nmin_expanded_in</th><td>1                                           </td><td>1                                   </td><td>1                                   </td><td>1                       </td><td>1                                   </td></tr>\n",
       "\t<tr><th scope=row>Nmin_compared_to</th><td>2                                           </td><td>1                                   </td><td>1                                   </td><td>1                       </td><td>1                                   </td></tr>\n",
       "\t<tr><th scope=row>min_expansion_factor</th><td>2                                           </td><td>2                                   </td><td>3                                   </td><td>2                       </td><td>2                                   </td></tr>\n",
       "\t<tr><th scope=row>expanded_in_all_found</th><td>YES                                         </td><td>NO                                  </td><td>YES                                 </td><td>NO                      </td><td>NO                                  </td></tr>\n",
       "\t<tr><th scope=row>compared_to_all_found</th><td>YES                                         </td><td>NO                                  </td><td>YES                                 </td><td>NO                      </td><td>NO                                  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 8 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X1 & X2 & X3 & X4 & X5\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tname & Expanded in Arabidopsis compared to Monocots & Expanded in barley compared to maize & bla                                  & bla2                     & bla3                                \\\\\n",
       "\texpanded\\_in & Arabidopsis\\_thaliana                         & Hordeum\\_vulgare                      & Zea\\_mays                             & Hordeum\\_vulgare;Zea\\_mays & Hordeum\\_vulgare;Zea\\_mays            \\\\\n",
       "\tcompared\\_to & Zea\\_mays;Hordeum\\_vulgare                     & Zea\\_mays                             & Hordeum\\_vulgare;Arabidopsis\\_thaliana & Arabidopsis\\_thaliana     & Hordeum\\_vulgare;Arabidopsis\\_thaliana\\\\\n",
       "\tNmin\\_expanded\\_in & 1                                            & 1                                    & 1                                    & 1                        & 1                                   \\\\\n",
       "\tNmin\\_compared\\_to & 2                                            & 1                                    & 1                                    & 1                        & 1                                   \\\\\n",
       "\tmin\\_expansion\\_factor & 2                                            & 2                                    & 3                                    & 2                        & 2                                   \\\\\n",
       "\texpanded\\_in\\_all\\_found & YES                                          & NO                                   & YES                                  & NO                       & NO                                  \\\\\n",
       "\tcompared\\_to\\_all\\_found & YES                                          & NO                                   & YES                                  & NO                       & NO                                  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 8 × 5\n",
       "\n",
       "| <!--/--> | X1 &lt;chr&gt; | X2 &lt;chr&gt; | X3 &lt;chr&gt; | X4 &lt;chr&gt; | X5 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| name | Expanded in Arabidopsis compared to Monocots | Expanded in barley compared to maize | bla                                  | bla2                     | bla3                                 |\n",
       "| expanded_in | Arabidopsis_thaliana                         | Hordeum_vulgare                      | Zea_mays                             | Hordeum_vulgare;Zea_mays | Hordeum_vulgare;Zea_mays             |\n",
       "| compared_to | Zea_mays;Hordeum_vulgare                     | Zea_mays                             | Hordeum_vulgare;Arabidopsis_thaliana | Arabidopsis_thaliana     | Hordeum_vulgare;Arabidopsis_thaliana |\n",
       "| Nmin_expanded_in | 1                                            | 1                                    | 1                                    | 1                        | 1                                    |\n",
       "| Nmin_compared_to | 2                                            | 1                                    | 1                                    | 1                        | 1                                    |\n",
       "| min_expansion_factor | 2                                            | 2                                    | 3                                    | 2                        | 2                                    |\n",
       "| expanded_in_all_found | YES                                          | NO                                   | YES                                  | NO                       | NO                                   |\n",
       "| compared_to_all_found | YES                                          | NO                                   | YES                                  | NO                       | NO                                   |\n",
       "\n"
      ],
      "text/plain": [
       "                      X1                                          \n",
       "name                  Expanded in Arabidopsis compared to Monocots\n",
       "expanded_in           Arabidopsis_thaliana                        \n",
       "compared_to           Zea_mays;Hordeum_vulgare                    \n",
       "Nmin_expanded_in      1                                           \n",
       "Nmin_compared_to      2                                           \n",
       "min_expansion_factor  2                                           \n",
       "expanded_in_all_found YES                                         \n",
       "compared_to_all_found YES                                         \n",
       "                      X2                                  \n",
       "name                  Expanded in barley compared to maize\n",
       "expanded_in           Hordeum_vulgare                     \n",
       "compared_to           Zea_mays                            \n",
       "Nmin_expanded_in      1                                   \n",
       "Nmin_compared_to      1                                   \n",
       "min_expansion_factor  2                                   \n",
       "expanded_in_all_found NO                                  \n",
       "compared_to_all_found NO                                  \n",
       "                      X3                                  \n",
       "name                  bla                                 \n",
       "expanded_in           Zea_mays                            \n",
       "compared_to           Hordeum_vulgare;Arabidopsis_thaliana\n",
       "Nmin_expanded_in      1                                   \n",
       "Nmin_compared_to      1                                   \n",
       "min_expansion_factor  3                                   \n",
       "expanded_in_all_found YES                                 \n",
       "compared_to_all_found YES                                 \n",
       "                      X4                      \n",
       "name                  bla2                    \n",
       "expanded_in           Hordeum_vulgare;Zea_mays\n",
       "compared_to           Arabidopsis_thaliana    \n",
       "Nmin_expanded_in      1                       \n",
       "Nmin_compared_to      1                       \n",
       "min_expansion_factor  2                       \n",
       "expanded_in_all_found NO                      \n",
       "compared_to_all_found NO                      \n",
       "                      X5                                  \n",
       "name                  bla3                                \n",
       "expanded_in           Hordeum_vulgare;Zea_mays            \n",
       "compared_to           Hordeum_vulgare;Arabidopsis_thaliana\n",
       "Nmin_expanded_in      1                                   \n",
       "Nmin_compared_to      1                                   \n",
       "min_expansion_factor  2                                   \n",
       "expanded_in_all_found NO                                  \n",
       "compared_to_all_found NO                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hypotheses_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 30 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>HOG</th><th scope=col>Zea_mays</th><th scope=col>Hordeum_vulgare</th><th scope=col>Arabidopsis_thaliana</th><th scope=col>exp_counter_Zea_mays</th><th scope=col>exp_counter_Hordeum_vulgare</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>N0.HOG0000014</td><td>74</td><td> 2</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000023</td><td>61</td><td> 2</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000027</td><td> 1</td><td>55</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000028</td><td>12</td><td>42</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000032</td><td> 1</td><td>47</td><td>4</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000047</td><td> 8</td><td>31</td><td>3</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000061</td><td> 0</td><td>36</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000073</td><td> 4</td><td>29</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000079</td><td> 2</td><td> 4</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000089</td><td> 3</td><td>25</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000090</td><td> 2</td><td>26</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000099</td><td> 4</td><td>23</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000100</td><td> 0</td><td>27</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000101</td><td> 2</td><td>25</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000108</td><td> 2</td><td>20</td><td>5</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000115</td><td>11</td><td>14</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000116</td><td> 2</td><td>22</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000145</td><td> 1</td><td>20</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000146</td><td> 1</td><td>20</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000162</td><td> 8</td><td> 9</td><td>4</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000163</td><td> 1</td><td>19</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000177</td><td> 1</td><td>18</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000178</td><td> 1</td><td>18</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000197</td><td> 2</td><td>16</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000210</td><td> 6</td><td> 9</td><td>3</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000211</td><td> 1</td><td>15</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000213</td><td> 2</td><td>10</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000214</td><td> 6</td><td> 9</td><td>3</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000216</td><td> 0</td><td>14</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000218</td><td> 3</td><td>12</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " HOG & Zea\\_mays & Hordeum\\_vulgare & Arabidopsis\\_thaliana & exp\\_counter\\_Zea\\_mays & exp\\_counter\\_Hordeum\\_vulgare\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t N0.HOG0000014 & 74 &  2 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000023 & 61 &  2 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000027 &  1 & 55 & 2 & 0 & 1\\\\\n",
       "\t N0.HOG0000028 & 12 & 42 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000032 &  1 & 47 & 4 & 0 & 1\\\\\n",
       "\t N0.HOG0000047 &  8 & 31 & 3 & 1 & 1\\\\\n",
       "\t N0.HOG0000061 &  0 & 36 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000073 &  4 & 29 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000079 &  2 &  4 & 2 & 0 & 1\\\\\n",
       "\t N0.HOG0000089 &  3 & 25 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000090 &  2 & 26 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000099 &  4 & 23 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000100 &  0 & 27 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000101 &  2 & 25 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000108 &  2 & 20 & 5 & 0 & 1\\\\\n",
       "\t N0.HOG0000115 & 11 & 14 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000116 &  2 & 22 & 2 & 0 & 1\\\\\n",
       "\t N0.HOG0000145 &  1 & 20 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000146 &  1 & 20 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000162 &  8 &  9 & 4 & 1 & 1\\\\\n",
       "\t N0.HOG0000163 &  1 & 19 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000177 &  1 & 18 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000178 &  1 & 18 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000197 &  2 & 16 & 1 & 1 & 1\\\\\n",
       "\t N0.HOG0000210 &  6 &  9 & 3 & 1 & 1\\\\\n",
       "\t N0.HOG0000211 &  1 & 15 & 2 & 0 & 1\\\\\n",
       "\t N0.HOG0000213 &  2 & 10 & 2 & 0 & 1\\\\\n",
       "\t N0.HOG0000214 &  6 &  9 & 3 & 1 & 1\\\\\n",
       "\t N0.HOG0000216 &  0 & 14 & 1 & 0 & 1\\\\\n",
       "\t N0.HOG0000218 &  3 & 12 & 1 & 1 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 6\n",
       "\n",
       "| HOG &lt;chr&gt; | Zea_mays &lt;dbl&gt; | Hordeum_vulgare &lt;dbl&gt; | Arabidopsis_thaliana &lt;dbl&gt; | exp_counter_Zea_mays &lt;dbl&gt; | exp_counter_Hordeum_vulgare &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| N0.HOG0000014 | 74 |  2 | 1 | 1 | 1 |\n",
       "| N0.HOG0000023 | 61 |  2 | 1 | 1 | 1 |\n",
       "| N0.HOG0000027 |  1 | 55 | 2 | 0 | 1 |\n",
       "| N0.HOG0000028 | 12 | 42 | 1 | 1 | 1 |\n",
       "| N0.HOG0000032 |  1 | 47 | 4 | 0 | 1 |\n",
       "| N0.HOG0000047 |  8 | 31 | 3 | 1 | 1 |\n",
       "| N0.HOG0000061 |  0 | 36 | 1 | 0 | 1 |\n",
       "| N0.HOG0000073 |  4 | 29 | 1 | 1 | 1 |\n",
       "| N0.HOG0000079 |  2 |  4 | 2 | 0 | 1 |\n",
       "| N0.HOG0000089 |  3 | 25 | 1 | 1 | 1 |\n",
       "| N0.HOG0000090 |  2 | 26 | 1 | 1 | 1 |\n",
       "| N0.HOG0000099 |  4 | 23 | 1 | 1 | 1 |\n",
       "| N0.HOG0000100 |  0 | 27 | 1 | 0 | 1 |\n",
       "| N0.HOG0000101 |  2 | 25 | 1 | 1 | 1 |\n",
       "| N0.HOG0000108 |  2 | 20 | 5 | 0 | 1 |\n",
       "| N0.HOG0000115 | 11 | 14 | 1 | 1 | 1 |\n",
       "| N0.HOG0000116 |  2 | 22 | 2 | 0 | 1 |\n",
       "| N0.HOG0000145 |  1 | 20 | 1 | 0 | 1 |\n",
       "| N0.HOG0000146 |  1 | 20 | 1 | 0 | 1 |\n",
       "| N0.HOG0000162 |  8 |  9 | 4 | 1 | 1 |\n",
       "| N0.HOG0000163 |  1 | 19 | 1 | 0 | 1 |\n",
       "| N0.HOG0000177 |  1 | 18 | 1 | 0 | 1 |\n",
       "| N0.HOG0000178 |  1 | 18 | 1 | 0 | 1 |\n",
       "| N0.HOG0000197 |  2 | 16 | 1 | 1 | 1 |\n",
       "| N0.HOG0000210 |  6 |  9 | 3 | 1 | 1 |\n",
       "| N0.HOG0000211 |  1 | 15 | 2 | 0 | 1 |\n",
       "| N0.HOG0000213 |  2 | 10 | 2 | 0 | 1 |\n",
       "| N0.HOG0000214 |  6 |  9 | 3 | 1 | 1 |\n",
       "| N0.HOG0000216 |  0 | 14 | 1 | 0 | 1 |\n",
       "| N0.HOG0000218 |  3 | 12 | 1 | 1 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "   HOG           Zea_mays Hordeum_vulgare Arabidopsis_thaliana\n",
       "1  N0.HOG0000014 74        2              1                   \n",
       "2  N0.HOG0000023 61        2              1                   \n",
       "3  N0.HOG0000027  1       55              2                   \n",
       "4  N0.HOG0000028 12       42              1                   \n",
       "5  N0.HOG0000032  1       47              4                   \n",
       "6  N0.HOG0000047  8       31              3                   \n",
       "7  N0.HOG0000061  0       36              1                   \n",
       "8  N0.HOG0000073  4       29              1                   \n",
       "9  N0.HOG0000079  2        4              2                   \n",
       "10 N0.HOG0000089  3       25              1                   \n",
       "11 N0.HOG0000090  2       26              1                   \n",
       "12 N0.HOG0000099  4       23              1                   \n",
       "13 N0.HOG0000100  0       27              1                   \n",
       "14 N0.HOG0000101  2       25              1                   \n",
       "15 N0.HOG0000108  2       20              5                   \n",
       "16 N0.HOG0000115 11       14              1                   \n",
       "17 N0.HOG0000116  2       22              2                   \n",
       "18 N0.HOG0000145  1       20              1                   \n",
       "19 N0.HOG0000146  1       20              1                   \n",
       "20 N0.HOG0000162  8        9              4                   \n",
       "21 N0.HOG0000163  1       19              1                   \n",
       "22 N0.HOG0000177  1       18              1                   \n",
       "23 N0.HOG0000178  1       18              1                   \n",
       "24 N0.HOG0000197  2       16              1                   \n",
       "25 N0.HOG0000210  6        9              3                   \n",
       "26 N0.HOG0000211  1       15              2                   \n",
       "27 N0.HOG0000213  2       10              2                   \n",
       "28 N0.HOG0000214  6        9              3                   \n",
       "29 N0.HOG0000216  0       14              1                   \n",
       "30 N0.HOG0000218  3       12              1                   \n",
       "   exp_counter_Zea_mays exp_counter_Hordeum_vulgare\n",
       "1  1                    1                          \n",
       "2  1                    1                          \n",
       "3  0                    1                          \n",
       "4  1                    1                          \n",
       "5  0                    1                          \n",
       "6  1                    1                          \n",
       "7  0                    1                          \n",
       "8  1                    1                          \n",
       "9  0                    1                          \n",
       "10 1                    1                          \n",
       "11 1                    1                          \n",
       "12 1                    1                          \n",
       "13 0                    1                          \n",
       "14 1                    1                          \n",
       "15 0                    1                          \n",
       "16 1                    1                          \n",
       "17 0                    1                          \n",
       "18 0                    1                          \n",
       "19 0                    1                          \n",
       "20 1                    1                          \n",
       "21 0                    1                          \n",
       "22 0                    1                          \n",
       "23 0                    1                          \n",
       "24 1                    1                          \n",
       "25 1                    1                          \n",
       "26 0                    1                          \n",
       "27 0                    1                          \n",
       "28 1                    1                          \n",
       "29 0                    1                          \n",
       "30 1                    1                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 10 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>HOG</th><th scope=col>Zea_mays</th><th scope=col>Hordeum_vulgare</th><th scope=col>Arabidopsis_thaliana</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>N0.HOG0000003</td><td>140</td><td> 1</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000014</td><td> 74</td><td> 2</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000023</td><td> 61</td><td> 2</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000027</td><td>  1</td><td>55</td><td>2</td></tr>\n",
       "\t<tr><td>N0.HOG0000028</td><td> 12</td><td>42</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000032</td><td>  1</td><td>47</td><td>4</td></tr>\n",
       "\t<tr><td>N0.HOG0000047</td><td>  8</td><td>31</td><td>3</td></tr>\n",
       "\t<tr><td>N0.HOG0000073</td><td>  4</td><td>29</td><td>1</td></tr>\n",
       "\t<tr><td>N0.HOG0000079</td><td>  2</td><td> 4</td><td>2</td></tr>\n",
       "\t<tr><td>N0.HOG0000086</td><td> 14</td><td> 8</td><td>8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 10 × 4\n",
       "\\begin{tabular}{llll}\n",
       " HOG & Zea\\_mays & Hordeum\\_vulgare & Arabidopsis\\_thaliana\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t N0.HOG0000003 & 140 &  1 & 1\\\\\n",
       "\t N0.HOG0000014 &  74 &  2 & 1\\\\\n",
       "\t N0.HOG0000023 &  61 &  2 & 1\\\\\n",
       "\t N0.HOG0000027 &   1 & 55 & 2\\\\\n",
       "\t N0.HOG0000028 &  12 & 42 & 1\\\\\n",
       "\t N0.HOG0000032 &   1 & 47 & 4\\\\\n",
       "\t N0.HOG0000047 &   8 & 31 & 3\\\\\n",
       "\t N0.HOG0000073 &   4 & 29 & 1\\\\\n",
       "\t N0.HOG0000079 &   2 &  4 & 2\\\\\n",
       "\t N0.HOG0000086 &  14 &  8 & 8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 10 × 4\n",
       "\n",
       "| HOG &lt;chr&gt; | Zea_mays &lt;dbl&gt; | Hordeum_vulgare &lt;dbl&gt; | Arabidopsis_thaliana &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| N0.HOG0000003 | 140 |  1 | 1 |\n",
       "| N0.HOG0000014 |  74 |  2 | 1 |\n",
       "| N0.HOG0000023 |  61 |  2 | 1 |\n",
       "| N0.HOG0000027 |   1 | 55 | 2 |\n",
       "| N0.HOG0000028 |  12 | 42 | 1 |\n",
       "| N0.HOG0000032 |   1 | 47 | 4 |\n",
       "| N0.HOG0000047 |   8 | 31 | 3 |\n",
       "| N0.HOG0000073 |   4 | 29 | 1 |\n",
       "| N0.HOG0000079 |   2 |  4 | 2 |\n",
       "| N0.HOG0000086 |  14 |  8 | 8 |\n",
       "\n"
      ],
      "text/plain": [
       "   HOG           Zea_mays Hordeum_vulgare Arabidopsis_thaliana\n",
       "1  N0.HOG0000003 140       1              1                   \n",
       "2  N0.HOG0000014  74       2              1                   \n",
       "3  N0.HOG0000023  61       2              1                   \n",
       "4  N0.HOG0000027   1      55              2                   \n",
       "5  N0.HOG0000028  12      42              1                   \n",
       "6  N0.HOG0000032   1      47              4                   \n",
       "7  N0.HOG0000047   8      31              3                   \n",
       "8  N0.HOG0000073   4      29              1                   \n",
       "9  N0.HOG0000079   2       4              2                   \n",
       "10 N0.HOG0000086  14       8              8                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3537"
      ],
      "text/latex": [
       "3537"
      ],
      "text/markdown": [
       "3537"
      ],
      "text/plain": [
       "[1] 3537"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "HOG_tibble <- HOG_tibble_complete\n",
    "\n",
    "\n",
    "# read-in species.tsv - easy lookup of ploidy per species \n",
    "species_table <- read.delim(\"../config/species.tsv\", header = TRUE, sep = \"\\t\", row.names = \"species\")\n",
    "#species_table\n",
    "#species_table[expanded_in, \"ploidy\"]\n",
    "\n",
    "\n",
    "# via snakemake params - Nmin info for expanded/compared per HOG in hypothesis.tsv (hypothesis specific) are parsed \n",
    "# here read-in from table as example\n",
    "hypotheses_table <- read.delim(\"../config/hypotheses.tsv\", header = TRUE, sep = \"\\t\", row.names = \"hypothesis\")\n",
    "Nmin_expanded_in <- hypotheses_table[\"Nmin_expanded_in\",1]\n",
    "#Nmin_compared_to <- hypotheses_table[\"Nmin_compared_to\",1]\n",
    "Nmin_compared_to = 1\n",
    "\n",
    "# expansion factor also in hypotheses.tsv = hypothesis specific\n",
    "#expansion_factor <- as.numeric(hypotheses_table[\"min_expansion_factor\", 1])\n",
    "expansion_factor = 2\n",
    "\n",
    "expansion_difference = 6\n",
    "ploidy_normalization = \"NO\"\n",
    "\n",
    "for (e in expanded_in) {\n",
    "    # create expansion counter column for each expanded species\n",
    "    exp_species_name = paste0(\"exp_counter_\", e)\n",
    "    HOG_tibble <- HOG_tibble %>% tibble::add_column(!!(exp_species_name) := 0)\n",
    "\n",
    "    if (ploidy_normalization == \"YES\") {\n",
    "        for (c in c_t_species) {\n",
    "            HOG_tibble <- HOG_tibble %>% \n",
    "                            mutate(!!(exp_species_name) := \n",
    "                            dplyr::case_when(\n",
    "                                # although cases in which the ct species has no genes are ignored via \n",
    "                                # the multiplication in the 2nd step, this underlines that we do so\n",
    "                                # now with the second expansion distance criterium we should have it anyway\n",
    "                                get(c) > 0 &\n",
    "                                get(e)/species_table[e, \"ploidy\"]*2 >= expansion_factor*(get(c))/species_table[c, \"ploidy\"]*2 ~ get(exp_species_name) + 1,\n",
    "                                get(c) > 0 &\n",
    "                                get(e)/species_table[e, \"ploidy\"]*2 >= expansion_difference + (get(c))/species_table[c, \"ploidy\"]*2 ~ get(exp_species_name) + 1,\n",
    "                                TRUE ~ 0,\n",
    "                              )\n",
    "                        )  \n",
    "        }\n",
    "    }\n",
    "    #if the user did not set ploidy_normalization to \"YES\"\n",
    "    #here, no ploidy information is used to divide the number of genes per OG\n",
    "    else {\n",
    "        for (c in c_t_species) {\n",
    "            HOG_tibble <- HOG_tibble %>% \n",
    "                            mutate(!!(exp_species_name) := \n",
    "                            dplyr::case_when(\n",
    "                                get(c) > 0 &\n",
    "                                get(e) >= expansion_factor*(get(c)) ~ get(exp_species_name) + 1,\n",
    "                                get(c) > 0 &\n",
    "                                get(e) >= expansion_difference + get(c) ~ get(exp_species_name) + 1,\n",
    "                                TRUE ~ 0,\n",
    "                              )\n",
    "                        )\n",
    "        }\n",
    "    }\n",
    "       \n",
    "}\n",
    "\n",
    "\n",
    "HOG_tibble %>% \n",
    "  arrange(desc(exp_counter_Hordeum_vulgare)) %>% head(n=30)\n",
    "\n",
    "\n",
    "#print(HOG_tibble)\n",
    "\n",
    "# then perform summing over all exp_counter_ columns based on user choices\n",
    "# at least Nmin_expanded_in expanded species that are expanded in at least Nmin_compared_to compared_to species\n",
    "# create new column\n",
    "# for each exp_counter_* species check if value >= y\n",
    "# all passing exp_counter_* species are counted; if sum >= x retain HOG\n",
    "\n",
    "HOG_tibble <- HOG_tibble %>%\n",
    "  mutate(pass =\n",
    "    # selecting columns that we want to use use for subsequent function\n",
    "    select(., contains(\"exp_counter_\")) %>%\n",
    "    # use pmap on this subset to get a vector of min from each row\n",
    "    # dataframe is a list so pmap works on each element of the list; here, each row  \n",
    "    # we sum the occurences (per row!) of exp_counter_* cols that greater/equal to user cutoff \n",
    "    purrr::pmap_dbl(., ~ sum(c(...) >= Nmin_compared_to))\n",
    "  )\n",
    "\n",
    "\n",
    "# lastly, retain rows/HOGs that pass x cutoff = number of expanded species\n",
    "# & drop unnecessary columns\n",
    "HOG_tibble <- HOG_tibble %>%\n",
    "  filter(pass >= Nmin_expanded_in) %>%\n",
    "  select(-contains(\"exp_counter_\"), -pass)\n",
    "\n",
    "\n",
    "# additional optional filter1 - expanded_in gene family complete criterium\n",
    "expanded_in_genefam_complete = \"yes\"\n",
    "if (expanded_in_genefam_complete == \"yes\") {\n",
    "  HOG_tibble <- HOG_tibble %>%\n",
    "  # create column with per-row sum of expanded_in species that have at least 1 gene in the HOG\n",
    "  mutate(expanded_in_pass =\n",
    "    select(., contains(expanded_in)) %>%\n",
    "    purrr::pmap_dbl(., ~ sum(c(...) >= 1))\n",
    "  ) %>%\n",
    "  # only keep rows/HOGs in which at least 1 gene of each expanded_in species occurs\n",
    "  filter(expanded_in_pass >= length(expanded_in)) %>%\n",
    "  # remove expanded_in_pass column\n",
    "  select(-expanded_in_pass)\n",
    "}\n",
    "\n",
    "\n",
    "# additional optional filter2 - compared_to gene family complete criterium\n",
    "compared_to_genefam_complete = \"yes\"\n",
    "if (compared_to_genefam_complete == \"yes\") {\n",
    "  HOG_tibble <- HOG_tibble %>%\n",
    "  # create column with per-row sum of compared_to species that have at least 1 gene in the HOG\n",
    "  mutate(compared_to_pass =\n",
    "    select(., contains(compared_to)) %>%\n",
    "    purrr::pmap_dbl(., ~ sum(c(...) >= 1))\n",
    "  ) %>%\n",
    "  # only keep rows/HOGs in which at least 1 gene of each compared_to species occurs\n",
    "  filter(compared_to_pass >= length(compared_to)) %>%\n",
    "  # remove compared_to_pass column\n",
    "  select(-compared_to_pass)\n",
    "}\n",
    "\n",
    "\n",
    "expanded_HOGs <- HOG_tibble\n",
    "\n",
    "head(expanded_HOGs, n=10)\n",
    "nrow(expanded_HOGs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "expanded_HOGs[expanded_HOGs$HOG==\"N0.HOG0000027\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# based on filtering criteria create per hypothesis table with:\n",
    "# all HOGs and gene counts + additional column expansion (yes/no)\n",
    "# this is later used in the creation of final tea outputs to create a HOG level table per hypothesis\n",
    "#HOG_tibble_complete \n",
    "\n",
    "#create expansion vector with length of expandsion HOGs\n",
    "expansion <- replicate(nrow(expanded_HOGs), \"yes\")\n",
    "\n",
    "#HOG_tibble_complete  add_column(expanded_HOGs, expansion, .after = \"HOG\")\n",
    "\n",
    "#?full_join\n",
    "\n",
    "expansion_tibble <- full_join(HOG_tibble_complete, add_column(expanded_HOGs, expansion, .after = \"HOG\"), \n",
    "                       by = c(\"HOG\"), suffix = c(\"\", \".remove\")) %>% \n",
    "                           replace_na(list(expansion = \"no\")) %>%\n",
    "                               select(-c(ends_with(\".remove\"))) \n",
    "\n",
    "#expansion_tibble\n",
    "\n",
    "saveRDS(expansion_tibble, paste(\"tea/\", num, \"/expansion_tibble/expansion_tibble.rds\", sep = \"\"))\n",
    "\n",
    "\n",
    "\n",
    "#?write_file\n",
    "#paste(\"tea/\", num, \"/expansion_tibble/expansion.tibble\", sep = \"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#additional quotation mark as first element is introduced for some reason\n",
    "#(delim space after : isn't handled properly)\n",
    "#workaround, we get rid with trailing [-1]\n",
    "#exp_og_genes <- unlist(strsplit(orthogroups[orthogroups$OG == \"OG0005769\",]$genes, split = \" \"))[-1]\n",
    "#exp_og_genes\n",
    "#y <- dplyr::filter(all_BLAST_reformatted, qseqid_name %in% exp_og_genes | sseqid_name %in% exp_og_genes)\n",
    "#sort the results by evalue (ascending V11), bitscore (descending V12) and pident (descending 3)\n",
    "#w\n",
    "#w[order(w$evalue, -w$bitscore, -w$pident),]\n",
    "#y <- arrange(y, evalue, -bitscore, -pident)\n",
    "#y\n",
    "#y$qseqid_name\n",
    "#list_qseqid <- as.character(y$qseqid_name)\n",
    "#list_sseqid <- as.character(y$sseqid_name)\n",
    "#list_merged <- unique(c(list_qseqid, list_sseqid))\n",
    "#list_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for each gene/protein name in an interesting OG do (is there a file just with names per OG?):\n",
    "#check \"all_BLAST_reformatted\" for all entries including these names and create new dataframe/tibble\n",
    "# then, perform filtering and retain certain set of genes/proteins per OG analysis\n",
    "#  then, create .txt file per OG with these gene names\n",
    "## come-up with filter criteria to have better trees?\n",
    "## I could of course just keep everything and save the evalues, etc.; well, problem for later.. ;D\n",
    "####> output for snakemake? what about inidividual OG txt files, because starting here parallelisation can really impact\n",
    "\n",
    "#additional quotation mark as first element is introduced for some reason\n",
    "#(delim space after : isn't handled properly)\n",
    "#workaround, we get rid with trailing [-1]\n",
    "\n",
    "#for (i in expanded_OGs$Orthogroup) {\n",
    "#    exp_og_genes <- unlist(strsplit(orthogroups[orthogroups$OG == i,]$genes, split = \" \"))[-1]\n",
    "#    BLAST_hits_exp_og_genes <- dplyr::filter(all_BLAST_reformatted, \n",
    "#                                             qseqid_name %in% exp_og_genes | sseqid_name %in% exp_og_genes)\n",
    "#    sorted_BLAST_hits_exp_og_genes <- arrange(BLAST_hits_exp_og_genes, evalue, -bitscore, -pident)\n",
    "#    list_qseqid <- as.character(sorted_BLAST_hits_exp_og_genes$qseqid_name)\n",
    "#    list_sseqid <- as.character(sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "#    list_merged <- unique(c(list_qseqid, list_sseqid))\n",
    "#    write_lines(list_merged,\n",
    "#           paste(\"tea/\", num, \"/exp_OGs_proteinnames/proteinnames_\", i, \".txt\", sep = \"\"))\n",
    "#}\n",
    "\n",
    "#lastly create .check to know it's done\n",
    "#exp_OGs_proteinnames.check <- \"check\"\n",
    "#write_file(exp_OGs_proteinnames.check, paste(\"checks/tea/\", num, \"/exp_OGs_proteinnames.check\", sep = \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create genes column in ph_orthogroups file\n",
    "# row merge? - no unite function, really handy ;D\n",
    "\n",
    "head(ph_orthogroups %>% unite(\"genes\", all_of(all_species), sep =\", \", na.rm = TRUE, remove = TRUE))\n",
    "ref_ph_orthogroups <- ph_orthogroups %>% unite(\"genes\", all_of(all_species), sep =\", \", na.rm = TRUE, remove = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#exp_og_genes <- unlist(strsplit(ref_ph_orthogroups[ref_ph_orthogroups$HOG == \"N0.HOG0000783\",]$genes, split = \", \"))\n",
    "#BLAST_hits_exp_og_genes <- dplyr::filter(all_BLAST_reformatted, \n",
    "#                                             qseqid_name %in% as.vector(exp_og_genes) | \n",
    "#                                             sseqid_name %in% exp_og_genes)# |\n",
    "#                                              sseqid_name %in% \"HORVU1Hr1G015850\")\n",
    "#BLAST_hits_exp_og_genes\n",
    "#exp_og_genes\n",
    "#head(all_BLAST_reformatted[all_BLAST_reformatted$sseqid_name==\"HORVU1Hr1G015850\",])\n",
    "#BLAST_hits_exp_og_genes\n",
    "#?filter\n",
    "#dput(exp_og_genes)\n",
    "\n",
    "#sorted_BLAST_hits_exp_og_genes <- arrange(BLAST_hits_exp_og_genes, evalue, -bitscore, -pident)\n",
    "#    list_qseqid <- as.character(sorted_BLAST_hits_exp_og_genes$qseqid_name)\n",
    "#    list_sseqid <- as.character(sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "#    list_merged <- unique(c(list_qseqid, list_sseqid))\n",
    "#list_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### playaround\n",
    "##### adding a specific number of additional best blast hits to a HOG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_og_genes <- unlist(strsplit(ref_ph_orthogroups[ref_ph_orthogroups$HOG == expanded_HOGs$HOG[47],]$genes, \n",
    "                                        split = \", \"))\n",
    "#exp_og_genes\n",
    "BLAST_hits_exp_og_genes <- dplyr::filter(all_BLAST_reformatted, \n",
    "                                             qseqid_name %in% exp_og_genes | sseqid_name %in% exp_og_genes)\n",
    "sorted_BLAST_hits_exp_og_genes <- arrange(BLAST_hits_exp_og_genes, evalue, -bitscore, -pident)\n",
    "length(exp_og_genes)\n",
    "#sorted_BLAST_hits_exp_og_genes\n",
    "add_blast_hits <- 20\n",
    "HOG_set_extended <- length(exp_og_genes) + add_blast_hits \n",
    "\n",
    "\n",
    "# if we have more than \"add_blast_hits\" further blast hits; limit the set to HOG + \"add_blast_hits\" param\n",
    "#if (length(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)) >= HOG_set_extended) {\n",
    "    list_qseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)[1:HOG_set_extended])\n",
    "#}\n",
    "# if we have less than \"add_blast_hits\" further blast hits; just take the complete set\n",
    "#else {\n",
    "#    list_qseqid <- as.character(sorted_BLAST_hits_exp_og_genes$qseqid_name)\n",
    "#}\n",
    "\n",
    "length(list_qseqid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#exp_og_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "12.05.2021  \n",
    "- ok, so this is actually way less complicated because I can take advantage how vectors work in R\n",
    "- set1 represents qseqids and set2 represents sseqids\n",
    "- unique() on both vectors concatenated rowwise! will yield a non-redundant set, maintaing order of appearance in vectors\n",
    "- if we substract the set of HOG genes (represented by h), then the remaining set is the n next blast hits (in order of filtering)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "set1 <- c(\"a\", \"b\", \"c\", \"d\", \"e\")\n",
    "set2 <- c(\"b\", \"g\", \"c\", \"c\", \"f\")\n",
    "\n",
    "h <- c(\"a\", \"b\", \"c\")\n",
    "\n",
    "m <- na.omit(unique(c(rbind(set1, set2))))\n",
    "m\n",
    "#na.omit(unique(c(rbind(set2, set1))))\n",
    "\n",
    "#setdiff will remove duplicates - which we shouldn't have here anymore\n",
    "setdiff(m, h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "expanded_HOGs$HOG[20]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**retaining up to a specific number of additional BLAST hits + order and info of the BLAST**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_og_genes <- unlist(strsplit(ref_ph_orthogroups[ref_ph_orthogroups$HOG == expanded_HOGs$HOG[20],]$genes, \n",
    "                                        split = \", \"))\n",
    "\n",
    "BLAST_hits_exp_og_genes <- dplyr::filter(all_BLAST_reformatted, \n",
    "                                             qseqid_name %in% exp_og_genes | sseqid_name %in% exp_og_genes)\n",
    "\n",
    "sorted_BLAST_hits_exp_og_genes <- arrange(BLAST_hits_exp_og_genes, evalue, -bitscore, -pident)\n",
    "\n",
    "# sorted table of all HOG genes with their extended blast hits\n",
    "#sorted_BLAST_hits_exp_og_genes\n",
    "\n",
    "# set value to extend the BLAST hits\n",
    "add_blast_hits <- 40\n",
    "HOG_set_extended <- length(exp_og_genes) + add_blast_hits\n",
    "\n",
    "# inside of for loop create per HOG extended gene list based on the additional blast hits\n",
    "# if we have more than \"add_blast_hits\" further blast hits; limit the set to HOG + \"add_blast_hits\" param\n",
    "#if (length(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)) >= HOG_set_extended) {\n",
    "#   list_qseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)[1:HOG_set_extended])\n",
    "#}\n",
    "# if we have less than \"add_blast_hits\" further blast hits; just take the complete set\n",
    "#else {\n",
    "#    list_qseqid <- as.character(sorted_BLAST_hits_exp_og_genes$qseqid_name)\n",
    "#}\n",
    "\n",
    "#list_qseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)[1:HOG_set_extended])\n",
    "#list_sseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$sseqid_name)[1:HOG_set_extended])\n",
    "#list_merged <- na.omit(unique(c(list_qseqid, list_sseqid)))\n",
    "                            \n",
    "# ...\n",
    "# final object is called \"list_merged\"\n",
    "#list_qseqid\n",
    "#list_sseqid\n",
    "#list_merged\n",
    "\n",
    "# use tail() to access last element in vector\n",
    "#tail(list_merged, n=1)\n",
    "\n",
    "# get row number of last appearance of last element in ssequid column\n",
    "# in some instances not as straightforward and last hit is actually on the query side\n",
    "# then use last entry rownumber there as subset cutoff\n",
    "#last <- sorted_BLAST_hits_exp_og_genes %>%\n",
    "#  rowid_to_column() %>%\n",
    "#  filter(\n",
    "#    case_when(\n",
    "#      sseqid_name != tail(list_merged, n=1) ~ qseqid_name == tail(list_merged, n=1),\n",
    "#      T ~ sseqid_name == tail(list_merged, n=1)\n",
    "##    )\n",
    "#  ) %>%\n",
    "#  tail(n=1)\n",
    "#last\n",
    "\n",
    "\n",
    "# get gene name of last gene to be added based on number of add_blast_hits\n",
    "all_blast_genes <- na.omit(\n",
    "       unique(\n",
    "         c(\n",
    "           rbind(\n",
    "             sorted_BLAST_hits_exp_og_genes$qseqid_name, \n",
    "             sorted_BLAST_hits_exp_og_genes$sseqid_name\n",
    "           )\n",
    "         )\n",
    "       )\n",
    "     ) \n",
    "\n",
    "#length(all_blast_genes)\n",
    "\n",
    "\n",
    "#length(unique(exp_og_genes))\n",
    "#length(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name))\n",
    "\n",
    "# set of all extended blast hits (based on threshold) - vector of gene names (ordered!)\n",
    "# also nice: don't need a conditional since `%>% head(n = add_blast_hits)` will work,\n",
    "# even if add_blast_hits param is > setdiff(all_blast_genes, exp_og_genes) \n",
    "extended_blast_hits_genes <- setdiff(all_blast_genes, exp_og_genes) %>% head(n = add_blast_hits)\n",
    "#length(extended_blast_hits_genes)\n",
    "\n",
    "# non redundant set of gene names of HOG + n additional blast hits as defined in the user threshold\n",
    "HOG_and_ext_blast_hits_genes <- c(exp_og_genes, extended_blast_hits_genes)\n",
    "length(HOG_and_ext_blast_hits_genes)\n",
    "\n",
    "#create subset of sorted_BLAST_hits_exp_og_genes table in which only:\n",
    "# exp_og_genes & extended_blast_hits_genes are allowed to appear\n",
    "# this way we have cutoff for the nth best blast hit/gene but also keep all secondary hits\n",
    "HOG_and_ext_blast_hits_table <- sorted_BLAST_hits_exp_og_genes %>%\n",
    "                                        filter(qseqid_name %in% HOG_and_ext_blast_hits_genes) %>%\n",
    "                                        filter(sseqid_name %in% HOG_and_ext_blast_hits_genes)\n",
    "\n",
    "head(HOG_and_ext_blast_hits_table)\n",
    "\n",
    "# create a subset of the \"sorted_BLAST_hits_exp_og_genes\" dataframe based on the final extended gene list\n",
    "# take last gene of extended set, search sseqid until last appearance and cut dataframe there?..\n",
    "# just keep all vs all information - additonal filtering as part of the WebApp\n",
    "#subset_sorted_BLAST_hits_exp_og_genes <- sorted_BLAST_hits_exp_og_genes[1:last$rowid,]\n",
    "#subset_sorted_BLAST_hits_exp_og_genes\n",
    "\n",
    "\n",
    "\n",
    "# finally create object per HOG which stores the BLAST information\n",
    "# object@name_of_HOG <- prev. subsetted tibble of BLAST hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## define custom class\n",
    "# need a list object to hold all data of this class\n",
    "extended_BLAST_hits <- list()\n",
    "\n",
    "# class for extended BLAST hits info\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    nrow_table=\"numeric\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\")#,\n",
    "#                    genes_extend=\"tbl_df\",\n",
    "#                    genes_complete=\"tbl_df\")\n",
    "         )\n",
    "\n",
    "ext_B_hits <- new(\"extended_BLAST_hits\",\n",
    "      blast_table=subset_sorted_BLAST_hits_exp_og_genes,\n",
    "      nrow_table=nrow(subset_sorted_BLAST_hits_exp_og_genes),\n",
    "      num_genes_HOG=length(exp_og_genes),\n",
    "      num_genes_extend = length(\n",
    "                  unique(\n",
    "                    c(\n",
    "                      unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "                      unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "                      )\n",
    "                  )\n",
    "                ) - length(exp_og_genes),\n",
    "      num_genes_complete=length(\n",
    "                  unique(\n",
    "                    c(\n",
    "                      unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "                      unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "                      )\n",
    "                  )\n",
    "                ),\n",
    "      genes_HOG=as_tibble(exp_og_genes)#,\n",
    "#      genes_extend=as_tibble(unique(\n",
    "#                    c(\n",
    "#                      unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "#                      unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "#                      )) - exp_og_genes),\n",
    "#      genes_complete=as_tibble(unique(\n",
    "#                    c(\n",
    "#                      unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "#                      unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "#                      )))\n",
    "    )\n",
    "\n",
    "ext_B_hits <- list(ext_B_hits)\n",
    "names(ext_B_hits) <- paste0(expanded_HOGs$HOG[21])\n",
    "\n",
    "#x <- list(test)\n",
    "#names(x) <- paste0(expanded_HOGs$HOG[47])\n",
    "\n",
    "extended_BLAST_hits <- c(extended_BLAST_hits, ext_B_hits)\n",
    "\n",
    "summary(extended_BLAST_hits)\n",
    "str(extended_BLAST_hits)\n",
    "\n",
    "#ext_B_hits$N0.HOG0000250"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "extended_BLAST_hits$N0.HOG0000101@genes_HOG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now HOG based\n",
    "\n",
    "#for each gene/protein name in an interesting OG do (is there a file just with names per OG?):\n",
    "#check \"all_BLAST_reformatted\" for all entries including these names and create new dataframe/tibble\n",
    "# then, perform filtering and retain certain set of genes/proteins per OG analysis\n",
    "#  then, create .txt file per OG with these gene names\n",
    "## come-up with filter criteria to have better trees?\n",
    "## I could of course just keep everything and save the evalues, etc.; well, problem for later.. ;D\n",
    "####> output for snakemake? what about inidividual OG txt files, because starting here parallelisation can really impact\n",
    "\n",
    "#additional quotation mark as first element is introduced for some reason\n",
    "#(delim space after : isn't handled properly)\n",
    "#workaround, we get rid with trailing [-1]\n",
    "#not a problem anymore with N0.tsv table as input\n",
    "\n",
    "dir.create(paste(\"tea/\", num, \"/extended_BLAST_hits/\", sep = \"\"))\n",
    "\n",
    "# define the number of additional best blast hits to include in the follow-up analyses\n",
    "add_blast_hits <- 20\n",
    "\n",
    "## define custom class for extended blast hits\n",
    "# need a list object to hold all data of this class\n",
    "extended_BLAST_hits <- list()\n",
    "\n",
    "# class for extended BLAST hits info\n",
    "setClass(\"extended_BLAST_hits\", \n",
    "         slots=list(blast_table=\"tbl_df\",\n",
    "                    num_genes_HOG=\"numeric\",\n",
    "                    num_genes_extend=\"numeric\",\n",
    "                    num_genes_complete=\"numeric\",\n",
    "                    genes_HOG=\"tbl_df\",\n",
    "                    genes_extend_hits=\"tbl_df\")\n",
    "         )\n",
    "\n",
    "\n",
    "for (i in expanded_HOGs$HOG) {\n",
    "    # split = \", \" (instead of \" \") is super important; didn't see this bug at first but\n",
    "    # it led to most genes not being recognized due to additional \",\" at the end..! (see tests in prev. cell)\n",
    "    exp_og_genes <- unlist(strsplit(ref_ph_orthogroups[ref_ph_orthogroups$HOG == i,]$genes, split = \", \"))\n",
    "    BLAST_hits_exp_og_genes <- dplyr::filter(all_BLAST_reformatted, \n",
    "                                             qseqid_name %in% exp_og_genes | sseqid_name %in% exp_og_genes)\n",
    "    sorted_BLAST_hits_exp_og_genes <- arrange(BLAST_hits_exp_og_genes, evalue, -bitscore, -pident)\n",
    "    # add number of chosen additional best blast hits to size of HOG \n",
    "    HOG_set_extended <- length(exp_og_genes) + add_blast_hits\n",
    "    \n",
    "    ### first qseqid\n",
    "    ### if we have more than \"additional genes\" further blast hits; limit the set to HOG + \"additional genes\" param\n",
    "    #if (length(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)) >= HOG_set_extended) {\n",
    "    ##    list_qseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$qseqid_name)[1:HOG_set_extended])\n",
    "    ##}\n",
    "    ## if we have less than \"additional genes\" further blast hits; just take the complete set\n",
    "    ##else {\n",
    "    ##    list_qseqid <- as.character(sorted_BLAST_hits_exp_og_genes$qseqid_name)\n",
    "    ##}\n",
    "    ### now for sseqid\n",
    "    ## if we have more than \"additional genes\" further blast hits; limit the set to HOG + \"additional genes\" param\n",
    "    ##if (length(unique(sorted_BLAST_hits_exp_og_genes$sseqid_name)) >= HOG_set_extended) {\n",
    "    ##    list_sseqid <- as.character(unique(sorted_BLAST_hits_exp_og_genes$sseqid_name)[1:HOG_set_extended])\n",
    "    ##}\n",
    "    ## if we have less than \"additional genes\" further blast hits; just take the complete set\n",
    "    ##else {\n",
    "    ##    list_sseqid <- as.character(sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "    ##}\n",
    "    ##list_merged <- unique(c(list_qseqid, list_sseqid))\n",
    "    \n",
    "    # get gene name of last gene to be added based on number of add_blast_hits\n",
    "    all_blast_genes <- na.omit(\n",
    "       unique(\n",
    "         c(\n",
    "           rbind(\n",
    "             sorted_BLAST_hits_exp_og_genes$qseqid_name, \n",
    "             sorted_BLAST_hits_exp_og_genes$sseqid_name\n",
    "           )\n",
    "         )\n",
    "       )\n",
    "     ) \n",
    "    \n",
    "    # set of all extended blast hits (based on threshold) - vector of gene names (ordered!)\n",
    "    # also nice: don't need a conditional since `%>% head(n = add_blast_hits)` will work,\n",
    "    # even if add_blast_hits param is > setdiff(all_blast_genes, exp_og_genes) \n",
    "    extended_blast_hits_genes <- setdiff(all_blast_genes, exp_og_genes) %>% head(n = add_blast_hits)\n",
    "    # length(extended_blast_hits_genes)\n",
    "  \n",
    "    # non redundant set of gene names of HOG + n additional blast hits as defined in the user threshold\n",
    "    HOG_and_ext_blast_hits_genes <- c(exp_og_genes, extended_blast_hits_genes)\n",
    "\n",
    "    #create subset of sorted_BLAST_hits_exp_og_genes table in which only:\n",
    "    # exp_og_genes & extended_blast_hits_genes are allowed to appear\n",
    "    # this way we have cutoff for the nth best blast hit/gene but also keep all secondary hits\n",
    "    HOG_and_ext_blast_hits_table <- sorted_BLAST_hits_exp_og_genes %>%\n",
    "                                      filter(qseqid_name %in% HOG_and_ext_blast_hits_genes) %>%\n",
    "                                      filter(sseqid_name %in% HOG_and_ext_blast_hits_genes)\n",
    "    \n",
    "    write_lines(HOG_and_ext_blast_hits_genes,\n",
    "           paste(\"tea/\", num, \"/exp_OGs_proteinnames/proteinnames_\", i, \".txt\", sep = \"\"))\n",
    "    \n",
    "    \n",
    "    # also: create list of S4 object containing the blast hits and respective information\n",
    "    \n",
    "    ## use tail() to access last element in vector\n",
    "    ##tail(list_merged, n=1)\n",
    "\n",
    "    ## get row number of last appearance of last element in ssequid column\n",
    "    ## in some instances not as straightforward and last hit is actually on the query side\n",
    "    ## then use last entry rownumber there as subset cutoff\n",
    "    ##last <- sorted_BLAST_hits_exp_og_genes %>%\n",
    "    ##  rowid_to_column() %>%\n",
    "    ##  filter(\n",
    "    ##   case_when(\n",
    "    ##      sseqid_name != tail(list_merged, n=1) ~ qseqid_name == tail(list_merged, n=1),\n",
    "    ##      T ~ sseqid_name == tail(list_merged, n=1)\n",
    "    ##    )\n",
    "    ##  ) %>%\n",
    "    ##  tail(n=1)\n",
    "    \n",
    "    ## create a subset of the \"sorted_BLAST_hits_exp_og_genes\" dataframe based on the final extended gene list\n",
    "    ## take last gene of extended set, search sseqid until last appearance and cut dataframe there\n",
    "    ## just keep all vs all information - additonal filtering as part of the WebApp\n",
    "    ##subset_sorted_BLAST_hits_exp_og_genes <- sorted_BLAST_hits_exp_og_genes[1:last$rowid,]\n",
    "    \n",
    "    # for each exp. HOG create an extended_BLAST_hits S4 object and collect as part of list\n",
    "    ext_B_hits <- new(\"extended_BLAST_hits\",\n",
    "      blast_table=HOG_and_ext_blast_hits_table,\n",
    "    #  nrow_table=nrow(subset_sorted_BLAST_hits_exp_og_genes),\n",
    "      num_genes_HOG=length(exp_og_genes),\n",
    "      num_genes_extend = length(extended_blast_hits_genes),\n",
    "      #            unique(\n",
    "      #              c(\n",
    "      #                unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "      #                unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "      #                )\n",
    "      #            )\n",
    "      #          ) - length(exp_og_genes),\n",
    "      num_genes_complete=length(HOG_and_ext_blast_hits_genes),\n",
    "      #            unique(\n",
    "      #              c(\n",
    "      #                unique(subset_sorted_BLAST_hits_exp_og_genes$qseqid_name), \n",
    "      #                unique(subset_sorted_BLAST_hits_exp_og_genes$sseqid_name)\n",
    "      #                )\n",
    "      #            )\n",
    "      #          ),\n",
    "      genes_HOG=as_tibble(exp_og_genes),\n",
    "      genes_extend_hits=as_tibble(extended_blast_hits_genes)\n",
    "            )\n",
    "    # assign name based on name of the underlying expanded HOG\n",
    "    ext_B_hits <- list(ext_B_hits)\n",
    "    names(ext_B_hits) <- paste0(i)\n",
    "    \n",
    "    # append to list object\n",
    "    extended_BLAST_hits <- c(extended_BLAST_hits, ext_B_hits)\n",
    "}\n",
    "\n",
    "# save extended BLAST hits to hypothesis specific (\"num\") RDS file \n",
    "#-> to be read and used in final_tea_computation.R script\n",
    "saveRDS(extended_BLAST_hits, paste(\"tea/\", num, \"/extended_BLAST_hits/extended_BLAST_hits.RDS\", sep = \"\"))\n",
    "\n",
    "#lastly create .check to know it's done\n",
    "exp_OGs_proteinnames.check <- \"check\"\n",
    "write_file(exp_OGs_proteinnames.check, paste(\"checks/tea/\", num, \"/exp_OGs_proteinnames.check\", sep = \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "as_tibble(extended_blast_hits_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "extended_BLAST_hits$N0.HOG0000101\n",
    "length(extended_BLAST_hits)\n",
    "str(expanded_HOGs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "paste(\"checks/tea/\", num, \"/exp_OGs_proteinnames.check\", sep = \"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_OGs_proteinnames.check <- \"check\"\n",
    "write_file(exp_OGs_proteinnames.check, paste(\"checks/tea/\", num, \"/exp_OGs_proteinnames.check\", sep = \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#str(all_BLAST_reformatted)\n",
    "#?read_delim\n",
    "library(stringr)\n",
    "conversion <- conversion %>%  mutate(seqid = str_remove_all(seqid, \":\"))\n",
    "str(conversion)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#perhaps continue as dataframe; have to see how easy this is"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#anyways after that snakemake steps:\n",
    "#1)faSomeRecords - extract the fasta sequences of all proteins per interesting OG\n",
    "#2)muscle - multiple sequence alignment\n",
    "#3)gblocks - eliminates poorly aligned positions and divergent regions of DNA or protein alignments\n",
    "#4)fastTree - build trees"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#I need to play around a bit with the string to column matching\n",
    "#there are several solutions to this - see https://stackoverflow.com/questions/25923392/select-columns-based-on-string-match-dplyrselect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
